# 에러

Node.js에서 실행되는 애플리케이션은 일반적으로 네 가지 범주의 에러를 경험하게 됩니다:

Node.js에서 발생하는 모든 JavaScript 및 시스템 에러는 표준 JavaScript [`<Error>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) 클래스를 상속받거나, 해당 클래스의 인스턴스입니다. 이러한 에러는 최소한 해당 클래스에서 제공하는 속성을 보장합니다.


### Node.js error codes[#](https://nodejs.org/docs/latest/api/errors.html#nodejs-error-codes)





#### `ABORT_ERR`

추가된 버전: v15.0.0

이 에러는 일반적으로 `AbortController`를 사용하여 작업이 중단되었을 때 발생합니다.

`AbortSignal`을 사용하지 않는 API들은 일반적으로 이 코드와 함께 에러를 발생시키지 않습니다.

이 코드는 웹 플랫폼의 `AbortError`와 호환되기 위해 Node.js 에러에서 사용하는 일반적인 `ERR_*` 규칙을 따르지 않습니다.


#### `ERR_ACCESS_DENIED`

이 오류는 Node.js가 [권한 모델(Permission Model)](https://nodejs.org/docs/latest/api/permissions.html#permission-model)에 의해 제한된 리소스에 접근하려고 할 때 발생하는 특별한 타입의 오류입니다.


#### `ERR_AMBIGUOUS_ARGUMENT`[#](https://nodejs.org/docs/latest/api/errors.html#err_ambiguous_argument)

함수의 인자가 함수 시그니처를 오해할 수 있는 방식으로 사용될 때 발생합니다. 이 오류는 `node:assert` 모듈에서 `assert.throws(block, message)`의 `message` 파라미터가 `block`에서 던져진 에러 메시지와 일치할 때 발생합니다. 이 경우, 사용자가 `message`를 `block`이 던질 것으로 예상한 메시지가 아니라, `block`이 에러를 던지지 않을 때 `AssertionError`가 표시할 메시지로 오해했을 가능성이 높기 때문입니다.


#### `ERR_ARG_NOT_ITERABLE`[#](https://nodejs.org/docs/latest/api/errors.html#err_arg_not_iterable)

Node.js API에서 `for...of` 루프와 함께 사용할 수 있는 반복 가능한(iterable) 인자가 필요했지만, 제공되지 않았을 때 발생하는 에러입니다.


#### `ERR_ASSERTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_assertion)

`ERR_ASSERTION`은 Node.js가 발생해서는 안 되는 예외적인 논리 위반을 감지했을 때 발생하는 특별한 타입의 에러입니다. 이 에러는 주로 `node:assert` 모듈에 의해 발생합니다.


#### `ERR_ASYNC_CALLBACK`[#](https://nodejs.org/docs/latest/api/errors.html#err_async_callback)

`AsyncHooks` 콜백으로 함수가 아닌 것을 등록하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_ASYNC_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_async_type)

비동기 리소스의 타입이 유효하지 않을 때 발생하는 오류입니다. 공개된 임베더 API를 사용한다면, 여러분도 직접 타입을 정의할 수 있습니다.


#### `ERR_BROTLI_COMPRESSION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_brotli_compression_failed)

Brotli 스트림에 전달된 데이터가 압축에 실패했을 때 발생하는 오류입니다.


#### `ERR_BROTLI_INVALID_PARAM`

Brotli 스트림을 생성하는 과정에서 잘못된 파라미터 키가 전달되었습니다.


#### `ERR_BUFFER_CONTEXT_NOT_AVAILABLE`

Node.js 인스턴스와 연결되지 않은 JS 엔진 컨텍스트에서 애드온 또는 임베더 코드를 통해 Node.js `Buffer` 인스턴스를 생성하려고 시도했을 때 발생하는 오류입니다. `Buffer` 메서드에 전달된 데이터는 메서드가 반환될 때 이미 해제된 상태입니다.

이 오류가 발생했을 때, `Buffer` 인스턴스를 생성하는 대신 일반적인 `Uint8Array`를 생성하는 방법을 고려할 수 있습니다. `Uint8Array`는 결과 객체의 프로토타입만 다를 뿐이며, 모든 Node.js 코어 API에서 `Buffer` 대신 사용할 수 있습니다. 또한 `Uint8Array`는 모든 컨텍스트에서 사용 가능합니다.


#### `ERR_BUFFER_OUT_OF_BOUNDS`

`Buffer`의 범위를 벗어난 작업이 시도되었습니다.


#### `ERR_BUFFER_TOO_LARGE`

`Buffer`를 생성하려고 시도했지만, 허용된 최대 크기를 초과한 경우 발생하는 에러입니다.


#### `ERR_CANNOT_WATCH_SIGINT`[#](https://nodejs.org/docs/latest/api/errors.html#err_cannot_watch_sigint)

Node.js가 `SIGINT` 신호를 감지하지 못했습니다.


#### `ERR_CHILD_CLOSED_BEFORE_REPLY`

자식 프로세스가 부모 프로세스로부터 응답을 받기 전에 종료되었습니다.


#### `ERR_CHILD_PROCESS_IPC_REQUIRED`

자식 프로세스를 생성할 때 IPC 채널을 지정하지 않았을 때 발생하는 에러입니다.


#### `ERR_CHILD_PROCESS_STDIO_MAXBUFFER`[#](https://nodejs.org/docs/latest/api/errors.html#err_child_process_stdio_maxbuffer)

이 오류는 메인 프로세스가 자식 프로세스의 STDERR 또는 STDOUT에서 데이터를 읽으려고 할 때, 데이터의 길이가 `maxBuffer` 옵션으로 설정된 값보다 길 경우 발생합니다.


#### `ERR_CLOSED_MESSAGE_PORT`[#](https://nodejs.org/docs/latest/api/errors.html#err_closed_message_port)

`MessagePort` 인스턴스가 닫힌 상태에서 사용하려고 시도할 때 발생하는 오류입니다. 일반적으로 `.close()` 메서드가 호출된 후에 이 오류가 나타납니다.


#### `ERR_CONSOLE_WRITABLE_STREAM`[#](https://nodejs.org/docs/latest/api/errors.html#err_console_writable_stream)

`Console`이 `stdout` 스트림 없이 인스턴스화되었거나, `Console`이 쓰기 가능하지 않은 `stdout` 또는 `stderr` 스트림을 가지고 있을 때 발생합니다.


#### `ERR_CONSTRUCT_CALL_INVALID`[#](https://nodejs.org/docs/latest/api/errors.html#err_construct_call_invalid)

추가된 버전: v12.5.0

호출할 수 없는 클래스 생성자가 호출되었을 때 발생하는 오류입니다.


#### `ERR_CONSTRUCT_CALL_REQUIRED`[#](https://nodejs.org/docs/latest/api/errors.html#err_construct_call_required)

클래스의 생성자를 `new` 없이 호출했을 때 발생하는 오류입니다.


#### `ERR_CONTEXT_NOT_INITIALIZED`

API에 전달된 vm 컨텍스트가 아직 초기화되지 않았습니다. 이 오류는 컨텍스트 생성 중에 오류가 발생했을 때(그리고 잡혔을 때) 나타날 수 있습니다. 예를 들어, 메모리 할당이 실패하거나 컨텍스트 생성 시 최대 호출 스택 크기에 도달한 경우에 발생할 수 있습니다.


#### `ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_custom_engine_not_supported)

OpenSSL 엔진이 요청되었지만, 현재 사용 중인 OpenSSL 버전에서 지원되지 않는 경우 발생합니다. 이는 `clientCertEngine`이나 `privateKeyEngine`과 같은 TLS 옵션을 통해 요청되었을 가능성이 높습니다. 이 문제는 주로 컴파일 타임 플래그인 `OPENSSL_NO_ENGINE` 때문에 발생합니다.


#### `ERR_CRYPTO_ECDH_INVALID_FORMAT`

`crypto.ECDH()` 클래스의 `getPublicKey()` 메서드에 전달된 `format` 인자의 값이 유효하지 않을 때 발생하는 오류입니다.


#### `ERR_CRYPTO_ECDH_INVALID_PUBLIC_KEY`

`crypto.ECDH()` 클래스의 `computeSecret()` 메서드에 전달된 `key` 인자의 값이 유효하지 않습니다. 이는 공개 키가 타원 곡선의 범위를 벗어났음을 의미합니다.


#### `ERR_CRYPTO_ENGINE_UNKNOWN`

[`require('node:crypto').setEngine()`](https://nodejs.org/docs/latest/api/crypto.html#cryptosetengineengine-flags)에 잘못된 암호화 엔진 식별자가 전달되었을 때 발생하는 오류입니다.


#### `ERR_CRYPTO_FIPS_FORCED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_fips_forced)

`--force-fips` 커맨드라인 인자를 사용했지만, `node:crypto` 모듈에서 FIPS 모드를 활성화하거나 비활성화하려는 시도가 있었을 때 발생하는 오류입니다.


#### `ERR_CRYPTO_FIPS_UNAVAILABLE`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_fips_unavailable)

FIPS 모드를 활성화하거나 비활성화하려고 시도했지만, FIPS 모드를 사용할 수 없는 경우 발생하는 오류입니다.


#### `ERR_CRYPTO_HASH_FINALIZED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_hash_finalized)

[`hash.digest()`](https://nodejs.org/docs/latest/api/crypto.html#hashdigestencoding)가 여러 번 호출되었습니다. `hash.digest()` 메서드는 `Hash` 객체 인스턴스당 한 번만 호출해야 합니다.


#### `ERR_CRYPTO_HASH_UPDATE_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_hash_update_failed)

[`hash.update()`](https://nodejs.org/docs/latest/api/crypto.html#hashupdatedata-inputencoding) 호출이 어떤 이유로든 실패한 경우 발생합니다. 이 오류는 거의 발생하지 않습니다.


#### `ERR_CRYPTO_INCOMPATIBLE_KEY`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_incompatible_key)

제공된 암호화 키가 시도한 작업과 호환되지 않습니다.


#### `ERR_CRYPTO_INCOMPATIBLE_KEY_OPTIONS`

선택한 공개키 또는 개인키 인코딩이 다른 옵션과 호환되지 않습니다.


#### `ERR_CRYPTO_INITIALIZATION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_initialization_failed)

추가된 버전: v15.0.0

암호화 하위 시스템 초기화가 실패했습니다.


#### `ERR_CRYO_INVALID_AUTH_TAG`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_auth_tag)

추가된 버전: v15.0.0

유효하지 않은 인증 태그가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_COUNTER`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_counter)

추가된 버전: v15.0.0

카운터 모드(counter-mode) 암호화에 잘못된 카운터가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_CURVE`

추가된 버전: v15.0.0

유효하지 않은 타원 곡선이 제공되었습니다.


#### `ERR_CRYPTO_INVALID_DIGEST`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_digest)

유효하지 않은 [암호화 다이제스트 알고리즘](https://nodejs.org/docs/latest/api/crypto.html#cryptogethashes)이 지정되었습니다.


#### `ERR_CRYPTO_INVALID_IV`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_iv)

추가된 버전: v15.0.0

유효하지 않은 초기화 벡터(Initialization Vector, IV)가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_JWK`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_jwk)

추가된 버전: v15.0.0

유효하지 않은 JSON Web Key가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_KEYLEN`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_keylen)

추가된 버전: v15.0.0

유효하지 않은 키 길이가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_KEYPAIR`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_keypair)

추가된 버전: v15.0.0

유효하지 않은 키 쌍이 제공되었습니다.


#### `ERR_CRYPTO_INVALID_KEYTYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_keytype)

추가된 버전: v15.0.0

유효하지 않은 키 타입이 제공되었습니다.


#### `ERR_CRYPTO_INVALID_KEY_OBJECT_TYPE`

주어진 암호화 키 객체의 타입이 시도한 작업에 적합하지 않습니다.


#### `ERR_CRYPTO_INVALID_MESSAGELEN`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_messagelen)

추가된 버전: v15.0.0

유효하지 않은 메시지 길이가 제공되었습니다.


#### `ERR_CRYPTO_INVALID_SCRYPT_PARAMS`

추가된 버전: v15.0.0

[`crypto.scrypt()`](https://nodejs.org/docs/latest/api/crypto.html#cryptoscryptpassword-salt-keylen-options-callback) 또는 [`crypto.scryptSync()`](https://nodejs.org/docs/latest/api/crypto.html#cryptoscryptsyncpassword-salt-keylen-options) 함수의 하나 이상의 인자가 허용 범위를 벗어난 경우 발생합니다.


#### `ERR_CRYPTO_INVALID_STATE`

암호화 관련 메서드가 유효하지 않은 상태의 객체에 사용되었을 때 발생하는 오류입니다. 예를 들어, `cipher.final()`을 호출하기 전에 [`cipher.getAuthTag()`](https://nodejs.org/docs/latest/api/crypto.html#ciphergetauthtag)를 호출하는 경우가 이에 해당합니다.


#### `ERR_CRYPTO_INVALID_TAG_LENGTH`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_invalid_tag_length)

추가된 버전: v15.0.0

유효하지 않은 인증 태그 길이가 제공되었습니다.


#### `ERR_CRYPTO_JOB_INIT_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_job_init_failed)

추가된 버전: v15.0.0

비동기 암호화 작업 초기화가 실패했습니다.


#### `ERR_CRYPTO_JWK_UNSUPPORTED_CURVE`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_jwk_unsupported_curve)

키의 타원 곡선이 [JSON Web Key 타원 곡선 레지스트리](https://www.iana.org/assignments/jose/jose.xhtml#web-key-elliptic-curve)에 등록되어 있지 않아 발생하는 오류입니다.


#### `ERR_CRYPTO_JWK_UNSUPPORTED_KEY_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_jwk_unsupported_key_type)

사용하려는 비대칭 키 타입이 [JSON Web Key Types Registry](https://www.iana.org/assignments/jose/jose.xhtml#web-key-types)에 등록되어 있지 않아 발생하는 오류입니다.


#### `ERR_CRYPTO_OPERATION_FAILED`

추가된 버전: v15.0.0

암호화 작업이 특정되지 않은 이유로 실패했습니다.


#### `ERR_CRYPTO_PBKDF2_ERROR`

PBKDF2 알고리즘이 명시되지 않은 이유로 실패했습니다. OpenSSL은 더 자세한 정보를 제공하지 않기 때문에 Node.js도 추가적인 정보를 제공할 수 없습니다.


#### `ERR_CRYPTO_SCRYPT_NOT_SUPPORTED`

Node.js가 `scrypt` 지원 없이 컴파일되었습니다. 공식 릴리스 바이너리에서는 발생하지 않지만, 커스텀 빌드나 배포판 빌드에서 발생할 수 있습니다.


#### `ERR_CRYPTO_SIGN_KEY_REQUIRED`

[`sign.sign()`](https://nodejs.org/docs/latest/api/crypto.html#signsignprivatekey-outputencoding) 메서드에 서명을 위한 `key`가 제공되지 않았습니다.


#### `ERR_CRYPTO_TIMING_SAFE_EQUAL_LENGTH`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_timing_safe_equal_length)

[`crypto.timingSafeEqual()`](https://nodejs.org/docs/latest/api/crypto.html#cryptotimingsafeequala-b) 함수가 길이가 다른 `Buffer`, `TypedArray`, 또는 `DataView` 인자와 함께 호출되었습니다.


#### `ERR_CRYPTO_UNKNOWN_CIPHER`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_unknown_cipher)

알 수 없는 암호화 알고리즘이 지정되었습니다.


#### `ERR_CRYPTO_UNKNOWN_DH_GROUP`

알 수 없는 Diffie-Hellman 그룹 이름이 제공되었습니다. 유효한 그룹 이름 목록은 [`crypto.getDiffieHellman()`](https://nodejs.org/docs/latest/api/crypto.html#cryptogetdiffiehellmangroupname)을 참고하세요.


#### `ERR_CRYPTO_UNSUPPORTED_OPERATION`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_unsupported_operation)

추가된 버전: v15.0.0, v14.18.0

지원되지 않는 암호화 작업을 시도했을 때 발생하는 에러입니다.


#### `ERR_DEBUGGER_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_debugger_error)

추가된 버전: v16.4.0, v14.17.4

[디버거](https://nodejs.org/docs/latest/api/debugger.html)와 관련된 오류가 발생했습니다.


#### `ERR_DEBUGGER_STARTUP_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_debugger_startup_error)

추가된 버전: v16.4.0, v14.17.4

[디버거](https://nodejs.org/docs/latest/api/debugger.html)가 필요한 호스트/포트가 사용 가능해질 때까지 기다리다가 시간 초과가 발생했습니다.


#### `ERR_DIR_CLOSED`[#](https://nodejs.org/docs/latest/api/errors.html#err_dir_closed)

[`fs.Dir`](https://nodejs.org/docs/latest/api/fs.html#class-fsdir)가 이미 닫혀 있는 경우 발생하는 오류입니다.


#### `ERR_DIR_CONCURRENT_OPERATION`[#](https://nodejs.org/docs/latest/api/errors.html#err_dir_concurrent_operation)

추가된 버전: v14.3.0

비동기 작업이 진행 중인 [`fs.Dir`](https://nodejs.org/docs/latest/api/fs.html#class-fsdir)에 동기적인 읽기 또는 닫기 호출이 시도되었을 때 발생하는 오류입니다.


#### `ERR_DLOPEN_DISABLED`[#](https://nodejs.org/docs/latest/api/errors.html#err_dlopen_disabled)

추가된 버전: v16.10.0, v14.19.0

[`--no-addons`](https://nodejs.org/docs/latest/api/cli.html#--no-addons) 옵션을 사용하여 네이티브 애드온 로딩이 비활성화되었습니다.


#### `ERR_DLOPEN_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_dlopen_failed)

추가된 버전: v15.0.0

`process.dlopen()` 호출이 실패했을 때 발생하는 에러입니다.


#### `ERR_DNS_SET_SERVERS_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_dns_set_servers_failed)

`c-ares`가 DNS 서버를 설정하는 데 실패했습니다.


#### `ERR_DOMAIN_CALLBACK_NOT_AVAILABLE`

`node:domain` 모듈은 필요한 오류 처리 훅을 설정할 수 없어 사용할 수 없습니다. 이는 [`process.setUncaughtExceptionCaptureCallback()`](https://nodejs.org/docs/latest/api/process.html#processsetuncaughtexceptioncapturecallbackfn)이 이전 시점에 호출되었기 때문입니다.


#### `ERR_DOMAIN_CANNOT_SET_UNCAUGHT_EXCEPTION_CAPTURE`[#](https://nodejs.org/docs/latest/api/errors.html#err_domain_cannot_set_uncaught_exception_capture)

[`process.setUncaughtExceptionCaptureCallback()`](https://nodejs.org/docs/latest/api/process.html#processsetuncaughtexceptioncapturecallbackfn) 함수를 호출할 수 없는 이유는 `node:domain` 모듈이 이전 시점에 이미 로드되었기 때문입니다.

스택 트레이스에는 `node:domain` 모듈이 로드된 시점이 포함되어 있습니다.


#### `ERR_DUPLICATE_STARTUP_SNAPSHOT_MAIN_FUNCTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_duplicate_startup_snapshot_main_function)

[`v8.startupSnapshot.setDeserializeMainFunction()`](https://nodejs.org/docs/latest/api/v8.html#v8startupsnapshotsetdeserializemainfunctioncallback-data) 함수는 이미 한 번 호출된 상태에서 다시 호출하려고 할 때 발생하는 오류입니다. 이 함수는 한 번만 호출할 수 있기 때문에 중복 호출이 불가능합니다.


#### `ERR_ENCODING_INVALID_ENCODED_DATA`[#](https://nodejs.org/docs/latest/api/errors.html#err_encoding_invalid_encoded_data)

`TextDecoder()` API에 제공된 데이터가 지정된 인코딩에 따라 유효하지 않을 때 발생하는 오류입니다.


#### `ERR_ENCODING_NOT_SUPPORTED`

`TextDecoder()` API에 제공된 인코딩이 [WHATWG 지원 인코딩](https://nodejs.org/docs/latest/api/util.html#whatwg-supported-encodings) 목록에 포함되지 않았을 때 발생하는 오류입니다.


#### `ERR_EVAL_ESM_CANNOT_PRINT`[#](https://nodejs.org/docs/latest/api/errors.html#err_eval_esm_cannot_print)

`--print` 옵션은 ESM(ECMAScript 모듈) 입력과 함께 사용할 수 없습니다.


#### `ERR_EVENT_RECURSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_event_recURSION)

`EventTarget`에서 이벤트를 재귀적으로 발생시키려고 시도할 때 이 오류가 발생합니다.


#### `ERR_EXECUTION_ENVIRONMENT_NOT_AVAILABLE`

Node.js 환경이 JS 실행 컨텍스트와 연결되지 않았을 때 발생하는 오류입니다. 이 오류는 Node.js가 내장 라이브러리로 사용될 때, JS 엔진에 필요한 일부 훅이 제대로 설정되지 않은 경우에 나타날 수 있습니다.


#### `ERR_FALSY_VALUE_REJECTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_falsy_value_rejection)

`util.callbackify()`를 통해 콜백화된 `Promise`가 falsy 값으로 거부되었을 때 발생하는 에러입니다.


#### `ERR_FEATURE_UNAVAILABLE_ON_PLATFORM`[#](https://nodejs.org/docs/latest/api/errors.html#err_feature_unavailable_on_platform)

추가된 버전: v14.0.0

이 오류는 현재 실행 중인 Node.js 플랫폼에서 사용할 수 없는 기능을 사용하려고 할 때 발생합니다.


#### `ERR_FS_CP_DIR_TO_NON_DIR`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_dir_to_non_dir)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 디렉터리를 디렉터리가 아닌 대상(파일, 심볼릭 링크 등)에 복사하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_FS_CP_EEXIST`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_eexist)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 파일을 복사하려고 시도했을 때, 이미 존재하는 파일을 덮어쓰려고 했으며 `force`와 `errorOnExist` 옵션이 `true`로 설정된 경우 발생합니다.


#### `ERR_FS_CP_EINVAL`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_einval)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용할 때, `src` 또는 `dest`가 유효하지 않은 경로를 가리키는 경우 발생합니다.


#### `ERR_FS_CP_FIFO_PIPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_fifo_pipe)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 이름이 지정된 파이프를 복사하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_FS_CP_NON_DIR_TO_DIR`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_non_dir_to_dir)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 디렉터리가 아닌 파일이나 심볼릭 링크 등을 디렉터리로 복사하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_FS_CP_SOCKET`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_socket)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 소켓에 복사를 시도했을 때 발생하는 오류입니다.


#### `ERR_FS_CP_SYMLINK_TO_SUBDIRECTORY`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_symlink_to_subdirectory)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용할 때, `dest`에 있는 심볼릭 링크가 `src`의 하위 디렉토리를 가리키는 경우 발생합니다.


#### `ERR_FS_CP_UNKNOWN`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_cp_unknown)

추가된 버전: v16.7.0

[`fs.cp()`](https://nodejs.org/docs/latest/api/fs.html#fscpsrc-dest-options-callback)를 사용하여 알 수 없는 파일 타입을 복사하려고 시도했을 때 발생합니다.


#### `ERR_FS_EISDIR`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_eisdir)

경로가 디렉토리입니다.


#### `ERR_FS_FILE_TOO_LARGE`

`Buffer`가 허용하는 최대 크기보다 큰 파일을 읽으려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_ALTSVC_INVALID_ORIGIN`

HTTP/2 ALTSVC 프레임은 유효한 오리진(origin)을 필요로 합니다.


#### `ERR_HTTP2_ALTSVC_LENGTH`

HTTP/2 ALTSVC 프레임은 최대 16,382바이트의 페이로드 크기로 제한됩니다.


#### `ERR_HTTP2_CONNECT_AUTHORITY`

HTTP/2 요청에서 `CONNECT` 메서드를 사용할 때, `:authority` 의사 헤더가 필수입니다.


#### `ERR_HTTP2_CONNECT_PATH`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_connect_path)

HTTP/2 요청에서 `CONNECT` 메서드를 사용할 때, `:path` 의사 헤더(pseudo-header)는 허용되지 않습니다.


#### `ERR_HTTP2_CONNECT_SCHEME`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_connect_scheme)

HTTP/2 요청에서 `CONNECT` 메서드를 사용할 때, `:scheme` 의사 헤더(pseudo-header)는 허용되지 않습니다.


#### `ERR_HTTP2_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_error)

HTTP/2와 관련된 일반적인 오류가 발생했습니다.


#### `ERR_HTTP2_GOAWAY_SESSION`

`Http2Session`이 연결된 상대방으로부터 `GOAWAY` 프레임을 받은 후에는 새로운 HTTP/2 스트림을 열 수 없습니다.


#### `ERR_HTTP2_HEADERS_AFTER_RESPOND`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_headers_after_respond)

HTTP/2 응답이 시작된 후에 추가 헤더가 지정되었습니다.


#### `ERR_HTTP2_HEADERS_SENT`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_headers_sent)

여러 개의 응답 헤더를 보내려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_HEADER_SINGLE_VALUE`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_header_single_value)

HTTP/2 헤더 필드에 단일 값만 허용되는데 여러 값이 제공된 경우 발생하는 오류입니다.


#### `ERR_HTTP2_INFO_STATUS_NOT_ALLOWED`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_info_status_not_allowed)

HTTP/2 응답에서는 정보 제공용 HTTP 상태 코드(`1xx`)를 응답 상태 코드로 설정할 수 없습니다.


#### `ERR_HTTP2_INVALID_CONNECTION_HEADERS`

HTTP/2 요청과 응답에서는 HTTP/1 전용 헤더를 사용할 수 없습니다.


#### `ERR_HTTP2_INVALID_HEADER_VALUE`

유효하지 않은 HTTP/2 헤더 값이 지정되었습니다.


#### `ERR_HTTP2_INVALID_INFO_STATUS`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_invalid_info_status)

유효하지 않은 HTTP 정보 상태 코드가 지정되었습니다. 정보 상태 코드는 `100`부터 `199` 사이의 정수여야 합니다.


#### `ERR_HTTP2_INVALID_ORIGIN`

HTTP/2 `ORIGIN` 프레임은 유효한 오리진(origin)이 필요합니다.


#### `ERR_HTTP2_INVALID_PACKED_SETTINGS_LENGTH`

`http2.getUnpackedSettings()` API에 전달된 `Buffer`와 `Uint8Array` 인스턴스의 길이는 반드시 6의 배수여야 합니다.


#### `ERR_HTTP2_INVALID_PSEUDOHEADER`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_invalid_pseudoheader)

HTTP/2에서 유효한 의사 헤더(`:status`, `:path`, `:authority`, `:scheme`, `:method`)만 사용할 수 있습니다.


#### `ERR_HTTP2_INVALID_SESSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_invalid_session)

이미 종료된 `Http2Session` 객체에 대해 작업을 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_INVALID_SETTING_VALUE`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_invalid_setting_value)

HTTP/2 설정에 유효하지 않은 값이 지정되었습니다.


#### `ERR_HTTP2_INVALID_STREAM`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_invalid_stream)

이미 종료된 스트림에 대해 작업을 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_MAX_PENDING_SETTINGS_ACK`

HTTP/2 연결에서 `SETTINGS` 프레임을 상대방에게 보낼 때, 상대방은 이를 받고 적용했다는 확인 응답을 보내야 합니다. 기본적으로, 한 번에 보낼 수 있는 확인되지 않은 `SETTINGS` 프레임의 최대 개수가 정해져 있습니다. 이 오류 코드는 그 한계에 도달했을 때 발생합니다.


#### `ERR_HTTP2_NESTED_PUSH`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_nested_push)

푸시 스트림 내에서 새로운 푸시 스트림을 시작하려고 시도했습니다. 중첩된 푸시 스트림은 허용되지 않습니다.


#### `ERR_HTTP2_NO_MEM`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_no_mem)

`http2session.setLocalWindowSize(windowSize)` API를 사용할 때 메모리가 부족한 경우 발생하는 오류입니다.


#### `ERR_HTTP2_NO_SOCKET_MANIPULATION`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_no_socket_manipulation)

`Http2Session`에 연결된 소켓을 직접 조작(읽기, 쓰기, 일시 정지, 재개 등)하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_ORIGIN_LENGTH`

HTTP/2의 `ORIGIN` 프레임은 최대 16382바이트로 제한됩니다.


#### `ERR_HTTP2_OUT_OF_STREAMS`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_out_of_streams)

단일 HTTP/2 세션에서 생성된 스트림의 수가 최대 한계에 도달했습니다.


#### `ERR_HTTP2_PAYLOAD_FORBIDDEN`

HTTP 응답 코드에 대해 페이로드가 허용되지 않는 경우에 페이로드가 지정되었을 때 발생하는 오류입니다.


#### `ERR_HTTP2_PING_CANCEL`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_ping_cancel)

HTTP/2 핑이 취소되었습니다.


#### `ERR_HTTP2_PING_LENGTH`

HTTP/2 핑(ping) 페이로드의 길이는 정확히 8바이트여야 합니다.


#### `ERR_HTTP2_PSEUDOHEADER_NOT_ALLOWED`

HTTP/2 의사 헤더가 잘못 사용되었습니다. 의사 헤더는 `:` 접두사로 시작하는 헤더 키 이름입니다.


#### `ERR_HTTP2_PUSH_DISABLED`

클라이언트가 비활성화한 상태에서 푸시 스트림을 생성하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_SEND_FILE`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_send_file)

`Http2Stream.prototype.responseWithFile()` API를 사용하여 디렉토리를 전송하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP2_SEND_FILE_NOSEEK`

`Http2Stream.prototype.responseWithFile()` API를 사용하여 일반 파일이 아닌 다른 것을 보내려고 시도했지만, `offset` 또는 `length` 옵션이 제공된 경우 발생하는 오류입니다.


#### `ERR_HTTP2_SESSION_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_session_error)

`Http2Session`이 0이 아닌 오류 코드와 함께 종료되었습니다.


#### `ERR_HTTP2_SETTINGS_CANCEL`

`Http2Session` 설정이 취소되었습니다.


#### `ERR_HTTP2_SOCKET_BOUND`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_socket_bound)

`Http2Session` 객체를 `net.Socket`이나 `tls.TLSSocket`에 연결하려고 시도했지만, 해당 소켓이 이미 다른 `Http2Session` 객체에 바인딩된 경우 발생하는 오류입니다.


#### `ERR_HTTP2_SOCKET_UNBOUND`

이미 닫힌 `Http2Session`의 `socket` 속성을 사용하려고 시도할 때 발생하는 오류입니다.


#### `ERR_HTTP2_STATUS_101`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_status_101)

HTTP/2에서는 `101` 정보 상태 코드를 사용할 수 없습니다.


#### `ERR_HTTP2_STATUS_INVALID`

유효하지 않은 HTTP 상태 코드가 지정되었습니다. 상태 코드는 `100`에서 `599` 사이의 정수여야 합니다.


#### `ERR_HTTP2_STREAM_CANCEL`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_stream_cancel)

`Http2Stream`이 연결된 상대방에게 데이터를 전송하기 전에 소멸되었을 때 발생하는 오류입니다.


#### `ERR_HTTP2_STREAM_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_stream_error)

`RST_STREAM` 프레임에 0이 아닌 오류 코드가 지정되었습니다.


#### `ERR_HTTP2_STREAM_SELF_DEPENDENCY`

HTTP/2 스트림의 우선순위를 설정할 때, 스트림을 부모 스트림의 의존성으로 표시할 수 있습니다. 이 오류 코드는 스트림이 자기 자신을 의존성으로 표시하려고 시도할 때 발생합니다.


#### `ERR_HTTP2_TOO_MANY_CUSTOM_SETTINGS`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_too_many_custom_settings)

지원되는 커스텀 설정의 최대 개수(10개)를 초과했습니다.


#### `ERR_HTTP2_TOO_MANY_INVALID_FRAMES`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_too_many_invalid_frames)

추가된 버전: v15.14.0

`maxSessionInvalidFrames` 옵션으로 지정된 허용 가능한 잘못된 HTTP/2 프로토콜 프레임의 한계를 초과했습니다. 이 오류는 상대방이 보낸 잘못된 프레임이 너무 많을 때 발생합니다.


#### `ERR_HTTP2_TRAILERS_ALREADY_SENT`

`Http2Stream`에서 트레일러 헤더가 이미 전송되었습니다.


#### `ERR_HTTP2_TRAILERS_NOT_READY`

`http2stream.sendTrailers()` 메서드는 `Http2Stream` 객체에서 `'wantTrailers'` 이벤트가 발생한 후에만 호출할 수 있습니다. `'wantTrailers'` 이벤트는 `Http2Stream`에 `waitForTrailers` 옵션이 설정된 경우에만 발생합니다.


#### `ERR_HTTP2_UNSUPPORTED_PROTOCOL`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_unsupported_protocol)

`http2.connect()` 함수에 `http:` 또는 `https:` 이외의 프로토콜을 사용하는 URL이 전달되었습니다.


#### `ERR_HTTP_BODY_NOT_ALLOWED`

HTTP 응답에 내용을 허용하지 않는 경우, 이 에러가 발생합니다.


#### `ERR_HTTP_CONTENT_LENGTH_MISMATCH`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_content_length_mismatch)

응답 본문의 크기가 지정된 `Content-Length` 헤더 값과 일치하지 않을 때 발생하는 오류입니다.


#### `ERR_HTTP_HEADERS_SENT`

헤더가 이미 전송된 후에 추가 헤더를 보내려고 시도했을 때 발생하는 오류입니다.


#### `ERR_HTTP_INVALID_HEADER_VALUE`

유효하지 않은 HTTP 헤더 값이 지정되었습니다.


#### `ERR_HTTP_INVALID_STATUS_CODE`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_invalid_status_code)

상태 코드가 일반적인 상태 코드 범위(100-999)를 벗어난 경우 발생합니다.


#### `ERR_HTTP_REQUEST_TIMEOUT`

클라이언트가 허용된 시간 내에 전체 요청을 보내지 못한 경우 발생합니다.


#### `ERR_HTTP_SOCKET_ASSIGNED`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_socket_assigned)

주어진 [`ServerResponse`](https://nodejs.org/docs/latest/api/http.html#class-httpserverresponse)는 이미 소켓이 할당된 상태입니다.


#### `ERR_HTTP_SOCKET_ENCODING`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_socket_encoding)

[RFC 7230 Section 3](https://tools.ietf.org/html/rfc7230#section-3)에 따르면 소켓의 인코딩을 변경하는 것은 허용되지 않습니다.


#### `ERR_HTTP_TRAILER_INVALID`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_trailer_invalid)

`Trailer` 헤더가 설정되었지만, 현재 전송 인코딩 방식이 이를 지원하지 않는 경우 발생합니다.


#### `ERR_ILLEGAL_CONSTRUCTOR`[#](https://nodejs.org/docs/latest/api/errors.html#err_illegal_constructor)

이 오류는 공개되지 않은 생성자를 사용하여 객체를 생성하려고 시도했을 때 발생합니다.


#### `ERR_IMPORT_ATTRIBUTE_MISSING`

**추가된 버전:** v21.1.0

필요한 import 속성이 누락되어 지정된 모듈을 불러올 수 없습니다.


#### `ERR_IMPORT_ATTRIBUTE_TYPE_INCOMPATIBLE`[#](https://nodejs.org/docs/latest/api/errors.html#err_import_attribute_type_incompatible)

추가된 버전: v21.1.0

`import` 문에 `type` 속성이 지정되었지만, 해당 모듈의 타입이 일치하지 않을 때 발생하는 오류입니다.


#### `ERR_IMPORT_ATTRIBUTE_UNSUPPORTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_import_attribute_unsupported)

추가된 버전: v21.0.0, v20.10.0, v18.19.0

이 버전의 Node.js에서는 import 속성을 지원하지 않습니다.


#### `ERR_INCOMPATIBLE_OPTION_PAIR`[#](https://nodejs.org/docs/latest/api/errors.html#err_incompatible_option_pair)

두 옵션이 서로 호환되지 않아 동시에 사용할 수 없을 때 발생하는 오류입니다.


#### `ERR_INPUT_TYPE_NOT_ALLOWED`[#](https://nodejs.org/docs/latest/api/errors.html#err_input_type_not_allowed)

`--input-type` 플래그는 파일을 실행하려고 시도할 때 사용되었습니다. 이 플래그는 `--eval`, `--print`, 또는 `STDIN`을 통한 입력에서만 사용할 수 있습니다.


#### `ERR_INSPECTOR_ALREADY_ACTIVATED`

`node:inspector` 모듈을 사용할 때, 이미 특정 포트에서 동작 중인 inspector를 다시 활성화하려고 시도하면 이 오류가 발생합니다. 다른 주소에서 inspector를 활성화하기 전에 `inspector.close()`를 호출하여 현재 inspector를 종료해야 합니다.


#### `ERR_INSPECTOR_ALREADY_CONNECTED`

`node:inspector` 모듈을 사용할 때, 이미 연결된 상태에서 다시 연결을 시도하면 이 오류가 발생합니다.


#### `ERR_INSPECTOR_CLOSED`

`node:inspector` 모듈을 사용할 때, 이미 세션이 종료된 후에 인스펙터를 사용하려고 시도하면 이 에러가 발생합니다.


#### `ERR_INSPECTOR_COMMAND`[#](https://nodejs.org/docs/latest/api/errors.html#err_inspector_command)

`node:inspector` 모듈을 통해 커맨드를 실행하는 도중 오류가 발생했습니다.


#### `ERR_INSPECTOR_NOT_ACTIVE`

`inspector.waitForDebugger()`가 호출될 때 `inspector`가 활성화되어 있지 않으면 이 오류가 발생합니다.


#### `ERR_INSPECTOR_NOT_AVAILABLE`

`node:inspector` 모듈을 사용할 수 없습니다.


#### `ERR_INSPECTOR_NOT_CONNECTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_inspector_not_connected)

`node:inspector` 모듈을 사용할 때, 인스펙터가 연결되기 전에 사용하려고 시도하면 이 오류가 발생합니다.


#### `ERR_INSPECTOR_NOT_WORKER`[#](https://nodejs.org/docs/latest/api/errors.html#err_inspector_not_worker)

메인 스레드에서 호출된 API가 워커 스레드에서만 사용 가능한 경우 발생하는 오류입니다.


#### `ERR_INTERNAL_ASSERTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_internal_assertation)

이 오류는 Node.js 내부에 버그가 있거나 Node.js 내부 기능을 잘못 사용했을 때 발생합니다. 문제를 해결하려면 [https://github.com/nodejs/node/issues](https://github.com/nodejs/node/issues)에서 이슈를 열어주세요.


#### `ERR_INVALID_ADDRESS`

제공된 주소가 Node.js API에서 이해할 수 없는 형식입니다.


#### `ERR_INVALID_ADDRESS_FAMILY`

Node.js API가 제공된 주소 체계를 이해하지 못할 때 발생하는 오류입니다.


#### `ERR_INVALID_ARG_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_arg_type)

Node.js API에 잘못된 타입의 인자가 전달되었을 때 발생하는 오류입니다.


#### `ERR_INVALID_ARG_VALUE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_arg_value)

주어진 인자에 잘못되었거나 지원되지 않는 값이 전달되었을 때 발생하는 오류입니다.


#### `ERR_INVALID_ASYNC_ID`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_async_id)

`AsyncHooks`를 사용할 때 유효하지 않은 `asyncId` 또는 `triggerAsyncId`가 전달되었습니다. `-1`보다 작은 ID는 절대 발생하지 않아야 합니다.


#### `ERR_INVALID_BUFFER_SIZE`

`Buffer`에서 스왑(swap) 작업을 수행했지만, 해당 작업과 호환되지 않는 크기였을 때 발생하는 오류입니다.

```javascript
// 예제 코드
const buffer = Buffer.alloc(10); // 10바이트 크기의 버퍼 생성
buffer.swap16(); // 16비트 스왑 시도 (ERR_INVALID_BUFFER_SIZE 오류 발생)
```

위 예제에서 `swap16()` 메서드는 16비트(2바이트) 단위로 스왑을 수행합니다. 하지만 버퍼의 크기가 10바이트로, 2바이트의 배수가 아니기 때문에 `ERR_INVALID_BUFFER_SIZE` 오류가 발생합니다. 이 오류를 해결하려면 버퍼의 크기를 2바이트의 배수로 조정해야 합니다.


#### `ERR_INVALID_CHAR`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_char)

헤더에 유효하지 않은 문자가 감지되었습니다.


#### `ERR_INVALID_CURSOR_POS`

특정 스트림의 커서는 컬럼을 지정하지 않고는 특정 행으로 이동할 수 없습니다.


#### `ERR_INVALID_FD`

파일 디스크립터('fd')가 유효하지 않을 때 발생하는 오류입니다. 예를 들어, 파일 디스크립터가 음수 값인 경우에 이 오류가 발생할 수 있습니다.


#### `ERR_INVALID_FD_TYPE`

파일 디스크립터('fd') 타입이 유효하지 않습니다.


#### `ERR_INVALID_FILE_URL_HOST`

Node.js API에서 `file:` URL을 사용하는 경우(예: [`fs`](https://nodejs.org/docs/latest/api/fs.html) 모듈의 특정 함수들), 호환되지 않는 호스트가 포함된 파일 URL을 만났을 때 발생하는 에러입니다. 이 상황은 Unix 계열 시스템에서만 발생할 수 있으며, `localhost` 또는 빈 호스트만 지원됩니다.


#### `ERR_INVALID_FILE_URL_PATH`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_file_url_path)

Node.js API 중 `file:` URL을 사용하는 기능(예: [`fs`](https://nodejs.org/docs/latest/api/fs.html) 모듈의 특정 함수)이 호환되지 않는 경로를 가진 파일 URL을 만났을 때 발생하는 오류입니다. 경로를 사용할 수 있는지 여부를 결정하는 정확한 규칙은 플랫폼에 따라 다릅니다.


#### `ERR_INVALID_HANDLE_TYPE`

자식 프로세스와의 IPC 통신 채널을 통해 지원되지 않는 "핸들"을 보내려고 시도했을 때 발생하는 오류입니다. 자세한 내용은 [`subprocess.send()`](https://nodejs.org/docs/latest/api/child_process.html#subprocesssendmessage-sendhandle-options-callback)와 [`process.send()`](https://nodejs.org/docs/latest/api/process.html#processsendmessage-sendhandle-options-callback)를 참고하세요.


#### `ERR_INVALID_HTTP_TOKEN`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_http_token)

유효하지 않은 HTTP 토큰이 제공되었습니다.


#### `ERR_INVALID_IP_ADDRESS`

IP 주소가 유효하지 않습니다.


#### `ERR_INVALID_MIME_SYNTAX`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_mime_syntax)

MIME 문법이 유효하지 않습니다.


#### `ERR_INVALID_MODULE`

추가된 버전: v15.0.0, v14.18.0

존재하지 않거나 유효하지 않은 모듈을 로드하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_INVALID_MODULE_SPECIFIER`

가져오려는 모듈 문자열이 유효하지 않은 URL, 패키지 이름, 또는 패키지 하위 경로 지정자입니다.


#### `ERR_INVALID_OBJECT_DEFINED_PROPERTY`

객체의 프로퍼티에 잘못된 속성을 설정하려고 할 때 발생하는 오류입니다.


#### `ERR_INVALID_PACKAGE_CONFIG`

유효하지 않은 [`package.json`](https://nodejs.org/docs/latest/api/packages.html#nodejs-packagejson-field-definitions) 파일을 파싱하는 데 실패했습니다.


#### `ERR_INVALID_PACKAGE_TARGET`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_package_target)

`package.json` 파일의 [`"exports"`](https://nodejs.org/docs/latest/api/packages.html#exports) 필드에 모듈 해석을 시도할 때 유효하지 않은 타겟 매핑 값이 포함되어 있습니다.


#### `ERR_INVALID_PROTOCOL`

`http.request()` 함수에 잘못된 `options.protocol` 값이 전달되었을 때 발생하는 오류입니다.


#### `ERR_INVALID_REPL_EVAL_CONFIG`

[`REPL`](https://nodejs.org/docs/latest/api/repl.html) 설정에서 `breakEvalOnSigint`와 `eval` 옵션을 동시에 지정했는데, 이는 지원되지 않는 조합입니다.


#### `ERR_INVALID_REPL_INPUT`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_repl_input)

입력값이 [`REPL`](https://nodejs.org/docs/latest/api/repl.html)에서 사용될 수 없을 때 발생하는 오류입니다. 이 오류가 발생하는 조건은 [`REPL`](https://nodejs.org/docs/latest/api/repl.html) 문서에 자세히 설명되어 있습니다.


#### `ERR_INVALID_RETURN_PROPERTY`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_return_property)

이 오류는 함수가 실행될 때 반환된 객체의 속성 중 하나에 유효하지 않은 값이 제공된 경우 발생합니다.


#### `ERR_INVALID_RETURN_PROPERTY_VALUE`

이 오류는 함수가 실행될 때 반환된 객체의 속성 중 하나가 예상된 타입의 값을 제공하지 않을 경우 발생합니다.


#### `ERR_INVALID_RETURN_VALUE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_return_value)

이 오류는 함수가 실행될 때 예상된 타입의 값을 반환하지 않을 경우 발생합니다. 예를 들어, 함수가 Promise를 반환해야 하는데 다른 타입의 값을 반환할 때 이 오류가 발생할 수 있습니다.


#### `ERR_INVALID_STATE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_state)

추가된 버전: v15.0.0

이 오류는 유효하지 않은 상태로 인해 작업을 완료할 수 없음을 나타냅니다. 예를 들어, 객체가 이미 파괴되었거나 다른 작업을 수행 중일 수 있습니다.

```javascript
// 예제 코드
const obj = {
  destroyed: false,
  operationInProgress: false,
  performOperation() {
    if (this.destroyed) {
      throw new Error('ERR_INVALID_STATE: 객체가 이미 파괴되었습니다.');
    }
    if (this.operationInProgress) {
      throw new Error('ERR_INVALID_STATE: 다른 작업이 진행 중입니다.');
    }
    this.operationInProgress = true;
    // 작업 수행
    this.operationInProgress = false;
  },
  destroy() {
    this.destroyed = true;
  }
};

obj.performOperation(); // 정상적으로 작업 수행
obj.destroy();
obj.performOperation(); // ERR_INVALID_STATE 오류 발생
```

이 예제에서 `performOperation` 메서드는 객체가 파괴되었거나 다른 작업이 진행 중일 때 `ERR_INVALID_STATE` 오류를 발생시킵니다.


#### `ERR_INVALID_SYNC_FORK_INPUT`

비동기 fork에 stdio 입력으로 `Buffer`, `TypedArray`, `DataView`, 또는 `string`이 제공된 경우 발생하는 오류입니다. 자세한 내용은 [`child_process`](https://nodejs.org/docs/latest/api/child_process.html) 모듈 문서를 참고하세요.


#### `ERR_INVALID_THIS`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_this)

Node.js API 함수가 호출될 때 호환되지 않는 `this` 값이 사용된 경우 발생합니다.

```js
const urlSearchParams = new URLSearchParams('foo=bar&baz=new');
const buf = Buffer.alloc(1);
urlSearchParams.has.call(buf, 'foo'); // 'ERR_INVALID_THIS' 코드와 함께 TypeError가 발생합니다.
```


#### `ERR_INVALID_TUPLE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_tuple)

[WHATWG](https://nodejs.org/docs/latest/api/url.html#the-whatwg-url-api) [`URLSearchParams` 생성자](https://nodejs.org/docs/latest/api/url.html#new-urlsearchparamsiterable)에 제공된 `iterable`의 요소가 `[name, value]` 튜플을 나타내지 않을 때 발생합니다. 즉, 요소가 반복 가능하지 않거나 정확히 두 개의 요소로 구성되지 않은 경우 이 에러가 발생합니다.


#### `ERR_INVALID_TYPESCRIPT_SYNTAX`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_typescript_syntax)

추가된 버전: v23.0.0

제공된 TypeScript 문법이 유효하지 않거나 지원되지 않습니다. 이 오류는 [타입 제거(type-stripping)](https://nodejs.org/docs/latest/api/typescript.html#type-stripping)를 통해 변환이 필요한 TypeScript 문법을 사용할 때 발생할 수 있습니다.


#### `ERR_INVALID_URI`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_uri)

유효하지 않은 URI가 전달되었습니다.


#### `ERR_INVALID_URL`

WHATWG [`URL` 생성자](https://nodejs.org/docs/latest/api/url.html#new-urlinput-base)나 레거시 [`url.parse()`](https://nodejs.org/docs/latest/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost)에 잘못된 URL이 전달되어 파싱에 실패한 경우 발생합니다. 이 오류 객체는 일반적으로 파싱에 실패한 URL을 포함하는 `'input'`이라는 추가 속성을 가지고 있습니다.


#### `ERR_INVALID_URL_SCHEME`

호환되지 않는 스키마(프로토콜)의 URL을 특정 목적으로 사용하려고 시도했을 때 발생하는 오류입니다. 이 오류는 현재 [`fs`](https://nodejs.org/docs/latest/api/fs.html) 모듈의 [WHATWG URL API](https://nodejs.org/docs/latest/api/url.html#the-whatwg-url-api) 지원에서만 사용되며, 여기서는 `'file'` 스키마를 가진 URL만 허용됩니다. 하지만 향후 다른 Node.js API에서도 사용될 수 있습니다.


#### `ERR_IPC_CHANNEL_CLOSED`[#](https://nodejs.org/docs/latest/api/errors.html#err_ipc_channel_closed)

이미 닫힌 IPC 통신 채널을 사용하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_IPC_DISCONNECTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_ipc_disconnected)

이미 연결이 끊어진 IPC 통신 채널을 다시 끊으려고 시도했을 때 발생하는 오류입니다. 자세한 내용은 [`child_process`](https://nodejs.org/docs/latest/api/child_process.html) 모듈 문서를 참고하세요.


#### `ERR_IPC_ONE_PIPE`

Node.js 자식 프로세스를 생성할 때 하나 이상의 IPC 통신 채널을 사용하려고 시도한 경우 발생하는 오류입니다. 자세한 내용은 [`child_process`](https://nodejs.org/docs/latest/api/child_process.html) 모듈 문서를 참고하세요.


#### `ERR_IPC_SYNC_FORK`

동기적으로 포크된 Node.js 프로세스와 IPC 통신 채널을 열려고 시도했을 때 발생하는 오류입니다. 자세한 내용은 [`child_process`](https://nodejs.org/docs/latest/api/child_process.html) 모듈 문서를 참고하세요.


#### `ERR_IP_BLOCKED`[#](https://nodejs.org/docs/latest/api/errors.html#err_ip_blocked)

IP가 `net.BlockList`에 의해 차단되었습니다.


#### `ERR_LOADER_CHAIN_INCOMPLETE`[#](https://nodejs.org/docs/latest/api/errors.html#err_loader_chain_incomplete)

추가된 버전: v18.6.0, v16.17.0

ESM 로더 훅이 `next()`를 호출하지 않고, 명시적으로 단락 신호를 보내지 않은 채 반환되었을 때 발생하는 오류입니다.


#### `ERR_LOAD_SQLITE_EXTENSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_load_sqlite_extension)

추가된 버전: v23.5.0

SQLite 확장 기능을 로드하는 동안 오류가 발생했습니다.


#### `ERR_MEMORY_ALLOCATION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_memory_allocation_failed)

메모리 할당 시도가 실패한 경우 발생하는 에러입니다. 주로 C++ 레이어에서 메모리를 할당하려고 할 때 실패하면 이 에러가 나타납니다.


#### `ERR_MESSAGE_TARGET_CONTEXT_UNAVAILABLE`

추가된 버전: v14.5.0, v12.19.0

[`MessagePort`](https://nodejs.org/docs/latest/api/worker_threads.html#class-messageport)에 전송된 메시지를 대상 [vm](https://nodejs.org/docs/latest/api/vm.html) `Context`에서 역직렬화할 수 없을 때 발생합니다. 현재 모든 Node.js 객체가 모든 컨텍스트에서 성공적으로 생성될 수 있는 것은 아니며, `postMessage()`를 사용해 전송하려고 하면 수신 측에서 실패할 수 있습니다.


#### `ERR_METHOD_NOT_IMPLEMENTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_method_not_implemented)

필요한 메서드가 구현되지 않았을 때 발생하는 오류입니다.


#### `ERR_MISSING_ARGS`

Node.js API에서 필수 인자가 전달되지 않았을 때 발생하는 에러입니다. 이 에러는 API 명세를 엄격히 준수하기 위해 사용됩니다. 어떤 경우에는 `func(undefined)`는 허용되지만 `func()`는 허용되지 않을 수 있습니다. 대부분의 네이티브 Node.js API에서는 `func(undefined)`와 `func()`가 동일하게 처리되며, 이 경우 [`ERR_INVALID_ARG_TYPE`](https://nodejs.org/docs/latest/api/errors.html#err_invalid_arg_type) 에러 코드가 대신 사용될 수 있습니다.


#### `ERR_MISSING_OPTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_missing_option)

옵션 객체를 받는 API에서 일부 옵션은 필수로 지정해야 합니다. 이 에러는 필수 옵션이 누락된 경우 발생합니다.


#### `ERR_MISSING_PASSPHRASE`

암호화된 키를 읽으려고 시도했지만, 암호를 지정하지 않았을 때 발생하는 오류입니다.


#### `ERR_MISSING_PLATFORM_FOR_WORKER`[#](https://nodejs.org/docs/latest/api/errors.html#err_missing_platform_for_worker)

현재 사용 중인 Node.js 인스턴스의 V8 플랫폼은 Worker 생성 기능을 지원하지 않습니다. 이 오류는 Worker를 지원하는 임베더(embedder)가 없기 때문에 발생합니다. 특히, 표준 빌드된 Node.js에서는 이 오류가 발생하지 않습니다.


#### `ERR_MODULE_NOT_FOUND`

ECMAScript 모듈 로더가 `import` 작업을 시도하거나 프로그램 진입점을 로드할 때 모듈 파일을 찾을 수 없는 경우 발생하는 오류입니다.


#### `ERR_MULTIPLE_CALLBACK`[#](https://nodejs.org/docs/latest/api/errors.html#err_multiple_callback)

콜백이 여러 번 호출되었을 때 발생하는 오류입니다.

콜백은 일반적으로 한 번만 호출되도록 설계됩니다. 쿼리가 성공하거나 실패할 수 있지만, 동시에 둘 다 일어날 수는 없기 때문입니다. 콜백을 여러 번 호출하면 이와 같은 상황이 발생할 수 있습니다.


#### `ERR_NAPI_CONS_FUNCTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_napi_cons_function)

`Node-API`를 사용할 때, 전달된 생성자가 함수가 아닌 경우 발생하는 오류입니다.


#### `ERR_NAPI_INVALID_DATAVIEW_ARGS`

`napi_create_dataview()` 함수를 호출할 때, 주어진 `offset`이 데이터 뷰의 범위를 벗어났거나 `offset + length`가 주어진 `buffer`의 길이보다 큰 경우 발생합니다.


#### `ERR_NAPI_INVALID_TYPEDARRAY_ALIGNMENT`

`napi_create_typedarray()` 함수를 호출할 때, 제공된 `offset` 값이 요소 크기의 배수가 아닌 경우 발생하는 오류입니다.


#### `ERR_NAPI_INVALID_TYPEDARRAY_LENGTH`

`napi_create_typedarray()`를 호출할 때, `(length * size_of_element) + byte_offset`가 주어진 `buffer`의 길이보다 큰 경우 발생합니다.


#### `ERR_NAPI_TSFN_CALL_JS`[#](https://nodejs.org/docs/latest/api/errors.html#err_napi_tsfn_call_js)

스레드 안전 함수(thread-safe function)의 자바스크립트 부분을 호출하는 동안 오류가 발생했습니다.


#### `ERR_NAPI_TSFN_GET_UNDEFINED`[#](https://nodejs.org/docs/latest/api/errors.html#err_napi_tsfn_get_undefined)

JavaScript의 `undefined` 값을 가져오려고 시도하는 동안 오류가 발생했습니다.


#### `ERR_NON_CONTEXT_AWARE_DISABLED`

컨텍스트 인식이 되지 않는 네이티브 애드온이 이를 허용하지 않는 프로세스에서 로드되었습니다.


#### `ERR_NOT_BUILDING_SNAPSHOT`[#](https://nodejs.org/docs/latest/api/errors.html#err_not_building_snapshot)

Node.js가 V8 시작 스냅샷을 생성하지 않는 상태에서, 스냅샷 생성 시에만 사용할 수 있는 작업을 시도했을 때 발생하는 오류입니다.


#### `ERR_NOT_IN_SINGLE_EXECUTABLE_APPLICATION`[#](https://nodejs.org/docs/latest/api/errors.html#err_not_in_single_executable_application)

추가된 버전: v21.7.0, v20.12.0

이 오류는 단일 실행 파일 애플리케이션(single-executable application)이 아닌 환경에서 해당 작업을 수행할 수 없을 때 발생합니다.


#### `ERR_NOT_SUPPORTED_IN_SNAPSHOT`[#](https://nodejs.org/docs/latest/api/errors.html#err_not_supported_in_snapshot)

시작 스냅샷을 빌드할 때 지원되지 않는 작업을 시도했을 때 발생하는 오류입니다.


#### `ERR_NO_CRYPTO`[#](https://nodejs.org/docs/latest/api/errors.html#err_no_crypto)

Node.js가 OpenSSL 암호화 기능을 지원하지 않도록 컴파일된 상태에서 암호화 기능을 사용하려고 시도했을 때 발생하는 에러입니다.


#### `ERR_NO_ICU`

[ICU](https://nodejs.org/docs/latest/api/intl.html#internationalization-support) 기능을 사용하려고 시도했지만, Node.js가 ICU 지원 없이 컴파일된 경우 발생하는 오류입니다.


#### `ERR_NO_TYPESCRIPT`

추가된 버전: v23.0.0

[네이티브 TypeScript 지원](https://nodejs.org/docs/latest/api/typescript.html#type-stripping)이 필요한 기능을 사용하려고 시도했지만, Node.js가 TypeScript 지원 없이 컴파일되었습니다.


#### `ERR_OPERATION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_operation_failed)

추가된 버전: v15.0.0

작업이 실패했습니다. 이 오류는 일반적으로 비동기 작업의 실패를 알리기 위해 사용됩니다.


#### `ERR_OUT_OF_RANGE`

주어진 값이 허용된 범위를 벗어났을 때 발생하는 오류입니다.


#### `ERR_PACKAGE_IMPORT_NOT_DEFINED`

`package.json` 파일의 [`"imports"`](https://nodejs.org/docs/latest/api/packages.html#imports) 필드에서 지정된 내부 패키지 스펙ifier 매핑이 정의되지 않았을 때 발생하는 오류입니다.


#### `ERR_PACKAGE_PATH_NOT_EXPORTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_package_path_not_exported)

`package.json` 파일의 [`"exports"`](https://nodejs.org/docs/latest/api/packages.html#exports) 필드에서 요청한 하위 경로가 내보내지지 않았을 때 발생하는 오류입니다. `exports`는 캡슐화되어 있기 때문에, 내보내지 않은 내부 모듈은 절대 URL을 사용하지 않는 한 패키지 해결 과정을 통해 가져올 수 없습니다.


#### `ERR_PARSE_ARGS_INVALID_OPTION_VALUE`

추가된 버전: v18.3.0, v16.17.0

`strict` 옵션이 `true`로 설정된 상태에서, [`util.parseArgs()`](https://nodejs.org/docs/latest/api/util.html#utilparseargsconfig) 함수가 [`<boolean>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type) 타입의 값을 기대하는 옵션에 [`<string>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#String_type) 타입의 값을 제공하거나, 반대로 [`<string>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#String_type) 타입의 값을 기대하는 옵션에 [`<boolean>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type) 타입의 값을 제공할 때 발생하는 에러입니다.


#### `ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL`[#](https://nodejs.org/docs/latest/api/errors.html#err_parse_args_unexpected_positional)

추가된 버전: v18.3.0, v16.17.0

[`util.parseArgs()`](https://nodejs.org/docs/latest/api/util.html#utilparseargsconfig)에서 `allowPositionals`가 `false`로 설정된 상태에서 위치 인자(positional argument)가 제공될 때 발생합니다.


#### `ERR_PARSE_ARGS_UNKNOWN_OPTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_parse_args_unknown_option)

추가된 버전: v18.3.0, v16.17.0

`strict`가 `true`로 설정된 상태에서 [`util.parseArgs()`](https://nodejs.org/docs/latest/api/util.html#utilparseargsconfig)가 실행될 때, `options`에 설정되지 않은 인자가 있으면 이 에러가 발생합니다.


#### `ERR_PERFORMANCE_INVALID_TIMESTAMP`

성능 마크(performance mark) 또는 측정(measure)에 유효하지 않은 타임스탬프 값이 제공되었습니다.


#### `ERR_PERFORMANCE_MEASURE_INVALID_OPTIONS`[#](https://nodejs.org/docs/latest/api/errors.html#err_performance_measure_invalid_options)

성능 측정을 위한 옵션이 잘못되었습니다.


#### `ERR_PROTO_ACCESS`[#](https://nodejs.org/docs/latest/api/errors.html#err_proto_access)

`Object.prototype.__proto__`에 접근하는 것은 [`--disable-proto=throw`](https://nodejs.org/docs/latest/api/cli.html#--disable-protomode) 옵션을 사용하여 금지되었습니다. 객체의 프로토타입을 가져오거나 설정하려면 [`Object.getPrototypeOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf)와 [`Object.setPrototypeOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf)를 사용해야 합니다.


#### `ERR_QUIC_APPLICATION_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_application_error)

추가된 버전: v23.4.0

QUIC 애플리케이션 오류가 발생했습니다.


#### `ERR_QUIC_CONNECTION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_connection_failed)

추가된 버전: v23.0.0

QUIC 연결을 설정하는 데 실패했습니다.


#### `ERR_QUIC_ENDPOINT_CLOSED`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_endpoint_closed)

추가된 버전: v23.0.0

QUIC 엔드포인트가 오류와 함께 닫혔습니다.


#### `ERR_QUIC_OPEN_STREAM_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_open_stream_failed)

추가된 버전: v23.0.0

QUIC 스트림을 열지 못했습니다.


#### `ERR_QUIC_TRANSPORT_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_transport_error)

추가된 버전: v23.4.0

QUIC 전송 오류가 발생했습니다.


#### `ERR_QUIC_VERSION_NEGOTIATION_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_quic_version_negotiation_error)

추가된 버전: v23.4.0

QUIC 세션이 버전 협상이 필요하여 실패했습니다.


#### `ERR_REQUIRE_ASYNC_MODULE`

[ES Module](https://nodejs.org/docs/latest/api/esm.html)을 `require()`로 불러오려고 할 때, 해당 모듈이 비동기식으로 동작하는 경우 발생합니다. 즉, 모듈 내부에 최상위 수준의 `await`가 포함되어 있습니다.

최상위 `await`가 어디에 있는지 확인하려면 `--experimental-print-required-tla` 옵션을 사용하세요. 이 옵션은 모듈을 실행한 후 최상위 `await`를 찾아줍니다.

```bash
node --experimental-print-required-tla your-script.js
```


#### `ERR_REQUIRE_CYCLE_MODULE`[#](https://nodejs.org/docs/latest/api/errors.html#err_require_cycle_module)

[ES Module](https://nodejs.org/docs/latest/api/esm.html)을 `require()`로 불러오려고 할 때, CommonJS에서 ESM으로 또는 ESM에서 CommonJS로의 순환 참조가 발생하면 이 오류가 나타납니다. ES Module은 이미 평가 중일 때 다시 평가할 수 없기 때문에 이러한 순환 참조는 허용되지 않습니다.

이 문제를 피하려면, 순환 참조에 관여하는 `require()` 호출이 ES Module(예: `createRequire()`를 통해) 또는 CommonJS 모듈의 최상위 레벨에서 발생하지 않도록 해야 합니다. 대신 내부 함수에서 지연 평가(lazy evaluation) 방식으로 처리해야 합니다.


#### `ERR_REQUIRE_ESM`[#](https://nodejs.org/docs/latest/api/errors.html#err_require_esm)

`require()` 함수를 사용하여 [ES Module](https://nodejs.org/docs/latest/api/esm.html)을 로드하려고 시도했을 때 발생하는 오류입니다.

이 오류는 현재 더 이상 사용되지 않습니다. `require()` 함수가 동기적인 ES 모듈 로드를 지원하기 때문입니다. 하지만 `require()`가 최상위 `await`를 포함하는 ES 모듈을 만나면, 대신 [`ERR_REQUIRE_ASYNC_MODULE`](https://nodejs.org/docs/latest/api/errors.html#err_require_async_module) 오류를 발생시킵니다.


#### `ERR_SCRIPT_EXECUTION_INTERRUPTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_script_execution_interrupted)

스크립트 실행이 `SIGINT`에 의해 중단되었습니다. (예: Ctrl+C를 눌렀을 때)


#### `ERR_SCRIPT_EXECUTION_TIMEOUT`

스크립트 실행 시간이 초과되었습니다. 이는 실행 중인 스크립트에 버그가 있을 가능성이 있습니다.


#### `ERR_SERVER_ALREADY_LISTEN`[#](https://nodejs.org/docs/latest/api/errors.html#err_server_already_listen)

`net.Server`가 이미 실행 중인 상태에서 [`server.listen()`](https://nodejs.org/docs/latest/api/net.html#serverlisten) 메서드가 호출되면 발생하는 에러입니다. 이 에러는 HTTP, HTTPS, HTTP/2 `Server` 인스턴스를 포함한 모든 `net.Server` 인스턴스에 적용됩니다.


#### `ERR_SERVER_NOT_RUNNING`

`net.Server`가 실행 중이지 않은 상태에서 [`server.close()`](https://nodejs.org/docs/latest/api/net.html#serverclosecallback) 메서드가 호출되었을 때 발생하는 에러입니다. 이 에러는 HTTP, HTTPS, HTTP/2 `Server` 인스턴스를 포함한 모든 `net.Server` 인스턴스에 적용됩니다.


#### `ERR_SINGLE_EXECUTABLE_APPLICATION_ASSET_NOT_FOUND`

추가된 버전: v21.7.0, v20.12.0

싱글 실행 가능 애플리케이션 API에 리소스를 식별하기 위해 키가 전달되었지만, 일치하는 항목을 찾을 수 없습니다.


#### `ERR_SOCKET_ALREADY_BOUND`

이미 바인딩된 소켓을 다시 바인딩하려고 시도했을 때 발생하는 에러입니다.


#### `ERR_SOCKET_BAD_BUFFER_SIZE`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_bad_buffer_size)

[`dgram.createSocket()`](https://nodejs.org/docs/latest/api/dgram.html#dgramcreatesocketoptions-callback) 함수에서 `recvBufferSize` 또는 `sendBufferSize` 옵션에 잘못된(음수) 크기가 전달되었을 때 발생하는 오류입니다.


#### `ERR_SOCKET_BAD_PORT`

API 함수가 0 이상 65536 미만의 포트 번호를 기대했지만, 유효하지 않은 값이 전달되었을 때 발생하는 오류입니다.


#### `ERR_SOCKET_BAD_TYPE`

소켓 타입(`udp4` 또는 `udp6`)을 기대하는 API 함수가 유효하지 않은 값을 받았을 때 발생하는 오류입니다.


#### `ERR_SOCKET_BUFFER_SIZE`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_buffer_size)

[`dgram.createSocket()`](https://nodejs.org/docs/latest/api/dgram.html#dgramcreatesocketoptions-callback)를 사용할 때, 수신 또는 송신 `Buffer`의 크기를 결정할 수 없는 경우 발생합니다.


#### `ERR_SOCKET_CLOSED`

이미 닫힌 소켓에 대해 작업을 시도했을 때 발생하는 오류입니다.


#### `ERR_SOCKET_CLOSED_BEFORE_CONNECTION`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_closed_before_connection)

연결 중인 소켓에서 [`net.Socket.write()`](https://nodejs.org/docs/latest/api/net.html#socketwritedata-encoding-callback)를 호출했을 때, 연결이 완료되기 전에 소켓이 닫히면 이 에러가 발생합니다.


#### `ERR_SOCKET_CONNECTION_TIMEOUT`

DNS에서 반환된 주소 중 어느 하나에도 소켓이 연결할 수 없어 시간 초과가 발생했습니다. 이 문제는 주소 선택 알고리즘이 자동으로 동작할 때 나타납니다.


#### `ERR_SOCKET_DGRAM_IS_CONNECTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_dgram_is_connected)

이미 연결된 소켓에서 [`dgram.connect()`](https://nodejs.org/docs/latest/api/dgram.html#socketconnectport-address-callback)를 호출했을 때 발생하는 에러입니다.


#### `ERR_SOCKET_DGRAM_NOT_CONNECTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_dgram_not_connected)

연결이 끊긴 소켓에서 [`dgram.disconnect()`](https://nodejs.org/docs/latest/api/dgram.html#socketdisconnect) 또는 [`dgram.remoteAddress()`](https://nodejs.org/docs/latest/api/dgram.html#socketremoteaddress)를 호출했을 때 발생하는 오류입니다.


#### `ERR_SOCKET_DGRAM_NOT_RUNNING`

UDP 서브시스템이 실행되지 않은 상태에서 호출이 발생했습니다.


#### `ERR_SOURCE_MAP_CORRUPT`[#](https://nodejs.org/docs/latest/api/errors.html#err_source_map_corrupt)

소스 맵이 존재하지 않거나 손상되어 파싱할 수 없습니다.


#### `ERR_SOURCE_MAP_MISSING_SOURCE`[#](https://nodejs.org/docs/latest/api/errors.html#err_source_map_missing_source)

소스 맵에서 가져온 파일을 찾을 수 없습니다.


#### `ERR_SQLITE_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_sqlite_error)

추가된 버전: v22.5.0

[SQLite](https://nodejs.org/docs/latest/api/sqlite.html)에서 반환된 오류입니다.


#### `ERR_SRI_PARSE`[#](https://nodejs.org/docs/latest/api/errors.html#err_sri_parse)

서브리소스 무결성(Subresource Integrity) 검사를 위해 제공된 문자열이 파싱되지 않았습니다. [서브리소스 무결성 명세](https://www.w3.org/TR/SRI/#the-integrity-attribute)를 참고하여 무결성 속성의 형식을 확인해 보세요.


#### `ERR_STREAM_ALREADY_FINISHED`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_already_finished)

스트림이 이미 종료된 상태에서 완료할 수 없는 스트림 메서드가 호출되었을 때 발생하는 오류입니다.


#### `ERR_STREAM_CANNOT_PIPE`

[`Writable`](https://nodejs.org/docs/latest/api/stream.html#class-streamwritable) 스트림에서 [`stream.pipe()`](https://nodejs.org/docs/latest/api/stream.html#readablepipedestination-options)를 호출하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_STREAM_DESTROYED`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_destroyed)

스트림이 `stream.destroy()`를 사용하여 파괴된 후, 해당 스트림에서 완료할 수 없는 메서드가 호출되었을 때 발생하는 오류입니다.


#### `ERR_STREAM_NULL_VALUES`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_null_values)

[`stream.write()`](https://nodejs.org/docs/latest/api/stream.html#writablewritechunk-encoding-callback)를 호출할 때 `null` 값을 전달하려고 시도한 경우 발생하는 에러입니다.


#### `ERR_STREAM_PREMATURE_CLOSE`

`stream.finished()`와 `stream.pipeline()`에서 반환되는 에러입니다. 이 에러는 스트림이나 파이프라인이 명시적인 에러 없이 비정상적으로 종료될 때 발생합니다.


#### `ERR_STREAM_PUSH_AFTER_EOF`

스트림에 `null`(EOF)이 푸시된 후에 [`stream.push()`](https://nodejs.org/docs/latest/api/stream.html#readablepushchunk-encoding)를 호출하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_STREAM_UNABLE_TO_PIPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_unable_to_pipe)

파이프라인에서 닫히거나 파괴된 스트림으로 파이프를 시도했을 때 발생하는 에러입니다.


#### `ERR_STREAM_UNSHIFT_AFTER_END_EVENT`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_unshift_after_end_event)

`'end'` 이벤트가 발생한 후에 [`stream.unshift()`](https://nodejs.org/docs/latest/api/stream.html#readableunshiftchunk-encoding)를 호출하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_STREAM_WRAP`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_wrap)

소켓에 문자열 디코더가 설정되었거나 디코더가 `objectMode`에 있을 때 중단을 방지합니다.

```js
const Socket = require('node:net').Socket;
const instance = new Socket();
instance.setEncoding('utf8');
```


#### `ERR_STREAM_WRITE_AFTER_END`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_write_after_end)

`stream.end()`가 호출된 후에 `stream.write()`를 호출하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_STRING_TOO_LONG`[#](https://nodejs.org/docs/latest/api/errors.html#err_string_too_long)

문자열을 생성하려고 시도했지만, 허용된 최대 길이를 초과한 경우 발생하는 오류입니다.


#### `ERR_SYNTHETIC`[#](https://nodejs.org/docs/latest/api/errors.html#err_synthetic)

진단 리포트를 위해 호출 스택을 캡처하기 위해 사용되는 인위적인 에러 객체입니다.


#### `ERR_SYSTEM_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_system_error)

Node.js 프로세스 내에서 명확하지 않거나 특정되지 않은 시스템 오류가 발생했습니다. 이 오류 객체는 추가적인 상세 정보를 담고 있는 `err.info` 객체 속성을 포함합니다.


#### `ERR_TAP_LEXER_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_tap_lexer_error)

이 오류는 **렉서(lexer)** 상태가 실패했음을 나타냅니다.


#### `ERR_TAP_PARSER_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_tap_parser_error)

이 오류는 파서가 실패한 상태를 나타냅니다. 오류를 발생시킨 토큰에 대한 추가 정보는 `cause` 속성을 통해 확인할 수 있습니다.


#### `ERR_TAP_VALIDATION_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_tap_validation_error)

이 오류는 TAP 검증이 실패했음을 나타냅니다.


#### `ERR_TEST_FAILURE`[#](https://nodejs.org/docs/latest/api/errors.html#err_test_failure)

이 오류는 테스트가 실패했음을 나타냅니다. 실패에 대한 추가 정보는 `cause` 속성을 통해 확인할 수 있습니다. `failureType` 속성은 실패가 발생했을 때 테스트가 수행 중이던 작업을 지정합니다.


#### `ERR_TLS_ALPN_CALLBACK_INVALID_RESULT`

이 오류는 `ALPNCallback`이 클라이언트가 제공한 ALPN 프로토콜 목록에 없는 값을 반환할 때 발생합니다.


#### `ERR_TLS_ALPN_CALLBACK_WITH_PROTOCOLS`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_alpn_callback_with_protocols)

이 오류는 `TLSServer`를 생성할 때 TLS 옵션에 `ALPNProtocols`와 `ALPNCallback`이 동시에 포함된 경우 발생합니다. 이 두 옵션은 서로 배타적입니다.


#### `ERR_TLS_CERT_ALTNAME_FORMAT`

이 오류는 사용자가 제공한 `subjectaltname` 속성이 인코딩 규칙을 위반할 경우 `checkServerIdentity`에 의해 발생합니다. Node.js 자체에서 생성된 인증서 객체는 항상 인코딩 규칙을 준수하므로 이 오류가 발생하지 않습니다.


#### `ERR_TLS_CERT_ALTNAME_INVALID`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_cert_altname_invalid)

TLS를 사용할 때, 상대방의 호스트 이름이나 IP가 해당 인증서의 `subjectAltNames` 목록과 일치하지 않는 경우 발생합니다.


#### `ERR_TLS_DH_PARAM_SIZE`

TLS를 사용할 때, Diffie-Hellman(`DH`) 키 교환 프로토콜에 제공된 매개변수가 너무 작습니다. 기본적으로 키 길이는 취약점을 피하기 위해 1024비트 이상이어야 합니다. 하지만 더 강력한 보안을 위해 2048비트 이상을 사용하는 것이 권장됩니다.


#### `ERR_TLS_HANDSHAKE_TIMEOUT`

TLS/SSL 핸드셰이크가 시간 초과되었습니다. 이 경우, 서버도 연결을 중단해야 합니다.


#### `ERR_TLS_INVALID_CONTEXT`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_invalid_context)

추가된 버전: v13.3.0

컨텍스트는 반드시 `SecureContext`여야 합니다.


#### `ERR_TLS_INVALID_PROTOCOL_METHOD`

지정된 `secureProtocol` 메서드가 유효하지 않습니다. 이 메서드는 알려지지 않았거나 보안상의 이유로 비활성화된 상태입니다.


#### `ERR_TLS_INVALID_PROTOCOL_VERSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_invalid_protocol_version)

유효한 TLS 프로토콜 버전은 `'TLSv1'`, `'TLSv1.1'`, `'TLSv1.2'`입니다.


#### `ERR_TLS_INVALID_STATE`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_invalid_state)

추가된 버전: v13.10.0, v12.17.0

TLS 소켓은 연결되어 있고 안전하게 설정되어야 합니다. 계속하기 전에 'secure' 이벤트가 발생했는지 확인하세요.


#### `ERR_TLS_PROTOCOL_VERSION_CONFLICT`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_protocol_version_conflict)

TLS 프로토콜의 `minVersion` 또는 `maxVersion`을 설정하려고 할 때, `secureProtocol`을 명시적으로 설정하려는 시도와 충돌이 발생합니다. 두 메커니즘 중 하나만 사용해야 합니다.


#### `ERR_TLS_PSK_SET_IDENTITY_HINT_FAILED`

PSK(Pre-Shared Key) identity hint 설정에 실패했습니다. 힌트가 너무 길 수 있습니다.


#### `ERR_TLS_RENEGOTIATION_DISABLED`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_renegotiation_disabled)

TLS 재협상이 비활성화된 소켓 인스턴스에서 TLS 재협상을 시도했을 때 발생하는 오류입니다.


#### `ERR_TLS_REQUIRED_SERVER_NAME`

TLS를 사용하는 동안 `server.addContext()` 메서드를 호출할 때 첫 번째 매개변수에 호스트 이름을 제공하지 않았습니다.


#### `ERR_TLS_SESSION_ATTACK`[#](https://nodejs.org/docs/latest/api/errors.html#err_tls_session_attack)

과도한 양의 TLS 재협상이 감지되었습니다. 이는 서비스 거부 공격(DoS)의 가능성이 있는 벡터입니다.


#### `ERR_TLS_SNI_FROM_SERVER`

TLS 서버 측 소켓에서 Server Name Indication(SNI)을 발급하려고 시도했지만, 이는 클라이언트 측에서만 유효한 동작입니다.


#### `ERR_TRACE_EVENTS_CATEGORY_REQUIRED`[#](https://nodejs.org/docs/latest/api/errors.html#err_trace_events_category_required)

`trace_events.createTracing()` 메서드를 사용할 때는 최소한 하나의 **트레이스 이벤트 카테고리**가 필요합니다.


#### `ERR_TRACE_EVENTS_UNAVAILABLE`[#](https://nodejs.org/docs/latest/api/errors.html#err_trace_events_unavailable)

`node:trace_events` 모듈을 로드할 수 없습니다. Node.js가 `--without-v8-platform` 플래그와 함께 컴파일되었기 때문입니다.


#### `ERR_TRANSFORM_ALREADY_TRANSFORMING`[#](https://nodejs.org/docs/latest/api/errors.html#err_transform_already_transforming)

`Transform` 스트림이 아직 변환 중인 상태에서 종료되었을 때 발생하는 오류입니다.


#### `ERR_TRANSFORM_WITH_LENGTH_0`[#](https://nodejs.org/docs/latest/api/errors.html#err_transform_with_length_0)

`Transform` 스트림이 아직 쓰기 버퍼에 데이터가 남아 있는 상태에서 종료되었습니다.


#### `ERR_TTY_INIT_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_tty_init_failed)

TTY 초기화가 시스템 오류로 인해 실패했습니다.


#### `ERR_UNAVAILABLE_DURING_EXIT`[#](https://nodejs.org/docs/latest/api/errors.html#err_unavailable_during_exit)

[`process.on('exit')`](https://nodejs.org/docs/latest/api/process.html#event-exit) 핸들러 내에서 호출되어서는 안 되는 함수가 호출되었을 때 발생하는 오류입니다.


#### `ERR_UNCAUGHT_EXCEPTION_CAPTURE_ALREADY_SET`[#](https://nodejs.org/docs/latest/api/errors.html#err_uncaught_exception_capture_already_set)

[`process.setUncaughtExceptionCaptureCallback()`](https://nodejs.org/docs/latest/api/process.html#processsetuncaughtexceptioncapturecallbackfn)이 두 번 호출되었지만, 콜백을 먼저 `null`로 재설정하지 않았습니다.

이 오류는 다른 모듈에서 등록한 콜백을 실수로 덮어쓰는 것을 방지하기 위해 설계되었습니다.


#### `ERR_UNESCAPED_CHARACTERS`[#](https://nodejs.org/docs/latest/api/errors.html#err_unescaped_characters)

이스케이프 처리되지 않은 문자를 포함한 문자열이 수신되었습니다.


#### `ERR_UNHANDLED_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_unhandled_error)

처리되지 않은 에러가 발생했을 때 나타나는 오류입니다. 예를 들어, [`EventEmitter`](https://nodejs.org/docs/latest/api/events.html#class-eventemitter)에서 `'error'` 이벤트가 발생했지만, 해당 이벤트를 처리할 핸들러가 등록되지 않은 경우에 이 오류가 발생합니다.


#### `ERR_UNKNOWN_BUILTIN_MODULE`

이 오류는 일반적으로 사용자 코드에서 발생하지 않는 특정한 종류의 내부 Node.js 오류를 식별하기 위해 사용됩니다. 이 오류의 인스턴스는 Node.js 바이너리 자체 내부의 버그를 가리킵니다.


#### `ERR_UNKNOWN_CREDENTIAL`

존재하지 않는 Unix 그룹 또는 사용자 식별자가 전달되었습니다.


#### `ERR_UNKNOWN_ENCODING`[#](https://nodejs.org/docs/latest/api/errors.html#err_unknown_encoding)

API에 잘못된 또는 알려지지 않은 인코딩 옵션이 전달되었을 때 발생합니다.


#### `ERR_UNKNOWN_FILE_EXTENSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_unknown_file_extension)

알려지지 않거나 지원되지 않는 파일 확장자를 가진 모듈을 로드하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_UNKNOWN_MODULE_FORMAT`[#](https://nodejs.org/docs/latest/api/errors.html#err_unknown_module_format)

알려지지 않거나 지원되지 않는 형식의 모듈을 로드하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_UNKNOWN_SIGNAL`

유효하지 않거나 알 수 없는 프로세스 신호가 유효한 신호를 기대하는 API에 전달되었을 때 발생합니다. 예를 들어, [`subprocess.kill()`](https://nodejs.org/docs/latest/api/child_process.html#subprocesskillsignal)과 같은 API가 이에 해당합니다.


#### `ERR_UNSUPPORTED_DIR_IMPORT`

디렉토리 URL을 `import`하는 것은 지원되지 않습니다. 대신, [패키지 이름을 사용하여 자기 참조](https://nodejs.org/docs/latest/api/packages.html#self-referencing-a-package-using-its-name)하고, [`package.json`](https://nodejs.org/docs/latest/api/packages.html#nodejs-packagejson-field-definitions) 파일의 [`"exports"`](https://nodejs.org/docs/latest/api/packages.html#exports) 필드에 [커스텀 서브 경로를 정의](https://nodejs.org/docs/latest/api/packages.html#subpath-exports)하세요.

```js
import './'; // 지원되지 않음
import './index.js'; // 지원됨
import 'package-name'; // 지원됨
```


#### `ERR_UNSUPPORTED_ESM_URL_SCHEME`

`file`과 `data` 이외의 URL 스키마를 사용한 `import`는 지원되지 않습니다.


#### `ERR_UNSUPPORTED_NODE_MODULES_TYPE_STRIPPING`

추가된 버전: v22.6.0

`node_modules` 디렉토리 하위에 있는 파일에 대해서는 타입 제거를 지원하지 않습니다.


#### `ERR_UNSUPPORTED_RESOLVE_REQUEST`[#](https://nodejs.org/docs/latest/api/errors.html#err_unsupported_resolve_request)

잘못된 모듈 참조를 해결하려고 시도할 때 발생하는 오류입니다. 이 오류는 다음과 같은 경우에 발생할 수 있습니다:

- `file` URL 스키마가 아닌 모듈에서 내장 모듈이 아닌 **bare specifier**를 임포트하거나 `import.meta.resolve()`를 호출할 때
- [특수 스키마](https://url.spec.whatwg.org/#special-scheme)가 아닌 URL 스키마를 가진 모듈에서 [상대 URL](https://url.spec.whatwg.org/#relative-url-string)을 사용할 때

```js
try {
    // `data:` URL 모듈에서 'bare-specifier' 패키지를 임포트하려고 시도
    await import('data:text/javascript,import "bare-specifier"');
} catch (e) {
    console.log(e.code); // ERR_UNSUPPORTED_RESOLVE_REQUEST
}
```


#### `ERR_USE_AFTER_CLOSE`[#](https://nodejs.org/docs/latest/api/errors.html#err_use_after_close)

이미 닫힌 것을 사용하려고 시도했을 때 발생하는 오류입니다.


#### `ERR_VALID_PERFORMANCE_ENTRY_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_valid_performance_entry_type)

Performance Timing API(`perf_hooks`)를 사용할 때 유효한 성능 항목 타입을 찾을 수 없는 경우 발생합니다.


#### `ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING`

동적 임포트 콜백이 지정되지 않았습니다.


#### `ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING_FLAG`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_dynamic_import_callback_missing_flag)

`--experimental-vm-modules` 플래그 없이 동적 임포트 콜백이 호출되었습니다.


#### `ERR_VM_MODULE_ALREADY_LINKED`

모듈을 링크하려고 시도했지만, 다음 이유 중 하나로 인해 링크가 불가능한 경우 발생합니다:

- 이미 링크가 완료된 경우 (`linkingStatus`가 `'linked'`)
- 현재 링크 중인 경우 (`linkingStatus`가 `'linking'`)
- 해당 모듈의 링크가 실패한 경우 (`linkingStatus`가 `'errored'`)


#### `ERR_VM_MODULE_CACHED_DATA_REJECTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_cached_data_rejected)

모듈 생성자에 전달된 `cachedData` 옵션이 유효하지 않습니다.


#### `ERR_VM_MODULE_CANNOT_CREATE_CACHED_DATA`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_cannot_create_cached_data)

이미 평가된 모듈에 대해 캐시 데이터를 생성할 수 없습니다.


#### `ERR_VM_MODULE_DIFFERENT_CONTEXT`

링커 함수에서 반환된 모듈이 부모 모듈과 다른 컨텍스트에서 생성되었습니다. 연결된 모듈은 반드시 동일한 컨텍스트를 공유해야 합니다.


#### `ERR_VM_MODULE_LINK_FAILURE`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_link_failure)

모듈을 연결하는 과정에서 실패가 발생했습니다.


#### `ERR_VM_MODULE_NOT_MODULE`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_not_module)

링크 작업이 완료된 Promise의 결과값이 `vm.Module` 객체가 아닐 때 발생하는 오류입니다.


#### `ERR_VM_MODULE_STATUS`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_status)

현재 모듈의 상태가 이 작업을 허용하지 않습니다. 이 오류의 구체적인 의미는 호출된 함수에 따라 다릅니다.


#### `ERR_WASI_ALREADY_STARTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_wasi_already_started)

WASI 인스턴스가 이미 시작되었습니다.


#### `ERR_WASI_NOT_STARTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_wasi_not_started)

WASI 인스턴스가 시작되지 않았습니다.


#### `ERR_WEBASSEMBLY_RESPONSE`[#](https://nodejs.org/docs/latest/api/errors.html#err_webassembly_response)

추가된 버전: v18.1.0

`WebAssembly.compileStreaming` 또는 `WebAssembly.instantiateStreaming`에 전달된 `Response`가 유효한 WebAssembly 응답이 아닙니다.


#### `ERR_WORKER_INIT_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_init_failed)

`Worker` 초기화가 실패했습니다.


#### `ERR_WORKER_INVALID_EXEC_ARGV`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_invalid_exec_argv)

`Worker` 생성자에 전달된 `execArgv` 옵션에 유효하지 않은 플래그가 포함되어 있습니다.


#### `ERR_WORKER_MESSAGING_ERRORED`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_messaging_errored)

**추가된 버전:** v22.5.0

목적지 스레드가 [`postMessageToThread()`](https://nodejs.org/docs/latest/api/worker_threads.html#workerpostmessagetothreadthreadid-value-transferlist-timeout)를 통해 전송된 메시지를 처리하는 도중 오류가 발생했습니다.


#### `ERR_WORKER_MESSAGING_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_messaging_failed)

추가된 버전: v22.5.0

[`postMessageToThread()`](https://nodejs.org/docs/latest/api/worker_threads.html#workerpostmessagetothreadthreadid-value-transferlist-timeout)에서 요청한 스레드가 유효하지 않거나 `workerMessage` 리스너가 없는 경우 발생합니다.


#### `ERR_WORKER_MESSAGING_SAME_THREAD`

**추가된 버전:** v22.5.0

[`postMessageToThread()`](https://nodejs.org/docs/latest/api/worker_threads.html#workerpostmessagetothreadthreadid-value-transferlist-timeout)에서 요청한 스레드 ID가 현재 스레드 ID와 동일할 때 발생하는 오류입니다.


#### `ERR_WORKER_MESSAGING_TIMEOUT`

**추가된 버전:** v22.5.0

[`postMessageToThread()`](https://nodejs.org/docs/latest/api/worker_threads.html#workerpostmessagetothreadthreadid-value-transferlist-timeout)를 통해 메시지를 보내는 과정에서 시간 초과가 발생했습니다.


#### `ERR_WORKER_NOT_RUNNING`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_not_running)

`Worker` 인스턴스가 현재 실행 중이 아니어서 작업이 실패했습니다.


#### `ERR_WORKER_OUT_OF_MEMORY`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_out_of_memory)

`Worker` 인스턴스가 메모리 한계에 도달하여 종료되었습니다.


#### `ERR_WORKER_PATH`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_path)

워커의 메인 스크립트 경로가 절대 경로도 아니고, `./` 또는 `../`로 시작하는 상대 경로도 아닌 경우 발생하는 에러입니다.


#### `ERR_WORKER_UNSERIALIZABLE_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_unserializable_error)

워커 스레드에서 발생한 예외를 직렬화하려는 모든 시도가 실패했을 때 발생하는 에러입니다.


#### `ERR_WORKER_UNSUPPORTED_OPERATION`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_unsupported_operation)

요청한 기능은 워커 스레드에서 지원되지 않습니다.


#### `ERR_ZLIB_INITIALIZATION_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_zlib_init_failed)

[`zlib`](https://nodejs.org/docs/latest/api/zlib.html) 객체 생성이 잘못된 설정으로 인해 실패했습니다.


#### `HPE_CHUNK_EXTENSIONS_OVERFLOW`[#](https://nodejs.org/docs/latest/api/errors.html#hpe_chunk_extensions_overflow)

추가된 버전: v21.6.2, v20.11.1, v18.19.1

청크 확장 데이터가 너무 많이 수신되었습니다. 악의적이거나 잘못 구성된 클라이언트로부터 보호하기 위해, 16 KiB 이상의 데이터가 수신되면 이 코드를 가진 `Error`가 발생합니다.


#### `HPE_HEADER_OVERFLOW`[#](https://nodejs.org/docs/latest/api/errors.html#hpe_header_overflow)

너무 많은 HTTP 헤더 데이터가 수신되었습니다. 악의적이거나 잘못 구성된 클라이언트로부터 보호하기 위해, `maxHeaderSize`보다 많은 HTTP 헤더 데이터가 수신되면 HTTP 파싱이 중단됩니다. 이 경우 요청이나 응답 객체가 생성되지 않고, 이 코드를 가진 `Error`가 발생합니다.


#### `HPE_UNEXPECTED_CONTENT_LENGTH`[#](https://nodejs.org/docs/latest/api/errors.html#hpe_unexpected_content_length)

서버가 `Content-Length` 헤더와 `Transfer-Encoding: chunked`를 동시에 보내고 있습니다.

`Transfer-Encoding: chunked`는 서버가 동적으로 생성된 콘텐츠를 위해 HTTP 지속적 연결을 유지할 수 있게 해줍니다. 이 경우, `Content-Length` HTTP 헤더를 사용할 수 없습니다.

`Content-Length` 또는 `Transfer-Encoding: chunked` 중 하나만 사용해야 합니다.


#### `MODULE_NOT_FOUND`[#](https://nodejs.org/docs/latest/api/errors.html#module_not_found)

CommonJS 모듈 로더가 [`require()`](https://nodejs.org/docs/latest/api/modules.html#requireid) 작업을 시도하거나 프로그램의 진입점을 로드할 때 모듈 파일을 찾을 수 없는 경우 발생합니다.


### 레거시 Node.js 오류 코드[#](https://nodejs.org/docs/latest/api/errors.html#legacy-nodejs-error-codes)

[안정성: 0](https://nodejs.org/docs/latest/api/documentation.html#stability-index) - 더 이상 사용되지 않음. 이 오류 코드들은 일관성이 없거나 제거되었습니다.


#### `ERR_CANNOT_TRANSFER_OBJECT`[#](https://nodejs.org/docs/latest/api/errors.html#err_cannot_transfer_object)

추가된 버전: v10.5.0  
제거된 버전: v12.5.0

`postMessage()`에 전달된 값이 전송을 지원하지 않는 객체를 포함하고 있을 때 발생하는 오류입니다.


#### `ERR_CPU_USAGE`[#](https://nodejs.org/docs/latest/api/errors.html#err_cpu_usage)

**제거된 버전:** v15.0.0

`process.cpuUsage`의 네이티브 호출을 처리할 수 없을 때 발생하는 오류입니다.


#### `ERR_CRYPTO_HASH_DIGEST_NO_UTF16`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_hash_digest_no_utf16)

추가된 버전: v9.0.0  
제거된 버전: v12.12.0

[`hash.digest()`](https://nodejs.org/docs/latest/api/crypto.html#hashdigestencoding) 메서드에서 UTF-16 인코딩을 사용하려고 시도했습니다. `hash.digest()` 메서드는 `encoding` 인자를 받아 `Buffer` 대신 문자열을 반환할 수 있지만, UTF-16 인코딩(예: `ucs` 또는 `utf16le`)은 지원되지 않습니다.


#### `ERR_CRYPTO_SCRYPT_INVALID_PARAMETER`[#](https://nodejs.org/docs/latest/api/errors.html#err_crypto_scrypt_invalid_parameter)

**제거된 버전:** v23.0.0

[`crypto.scrypt()`](https://nodejs.org/docs/latest/api/crypto.html#cryptoscryptpassword-salt-keylen-options-callback) 또는 [`crypto.scryptSync()`](https://nodejs.org/docs/latest/api/crypto.html#cryptoscryptsyncpassword-salt-keylen-options) 함수에 호환되지 않는 옵션 조합이 전달되었습니다. Node.js의 새로운 버전에서는 이 오류 코드 대신 [`ERR_INCOMPATIBLE_OPTION_PAIR`](https://nodejs.org/docs/latest/api/errors.html#err_incompatible_option_pair)를 사용하며, 이는 다른 API와 일관성을 유지합니다.


#### `ERR_FS_INVALID_SYMLINK_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_fs_invalid_symlink_type)

**제거된 버전:** v23.0.0

[`fs.symlink()`](https://nodejs.org/docs/latest/api/fs.html#fssymlinktarget-path-type-callback) 또는 [`fs.symlinkSync()`](https://nodejs.org/docs/latest/api/fs.html#fssymlinksynctarget-path-type) 메서드에 잘못된 심볼릭 링크 타입이 전달되었을 때 발생하는 오류입니다.


#### `ERR_HTTP2_FRAME_ERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_frame_error)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

HTTP/2 세션에서 개별 프레임을 전송하는 데 실패했을 때 발생합니다.


#### `ERR_HTTP2_HEADERS_OBJECT`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_headers_object)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 오류는 HTTP/2 헤더 객체가 필요한 상황에서 발생합니다.


#### `ERR_HTTP2_HEADER_REQUIRED`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_header_requried)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

HTTP/2 메시지에서 필수 헤더가 누락된 경우 발생하는 에러입니다.


#### `ERR_HTTP2_INFO_HEADERS_AFTER_RESPOND`

**추가된 버전:** v9.0.0  
**제거된 버전:** v10.0.0

HTTP/2 정보 헤더는 `Http2Stream.prototype.respond()` 메서드를 호출하기 **전에만** 보내야 합니다.


#### `ERR_HTTP2_STREAM_CLOSED`[#](https://nodejs.org/docs/latest/api/errors.html#err_http2_stream_closed)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 오류는 이미 닫힌 HTTP/2 스트림에 대해 작업을 수행하려고 할 때 발생합니다.


#### `ERR_HTTP_INVALID_CHAR`[#](https://nodejs.org/docs/latest/api/errors.html#err_http_invalid_char)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 오류는 HTTP 응답 상태 메시지(이유 구문)에 유효하지 않은 문자가 발견되었을 때 발생합니다.


#### `ERR_IMPORT_ASSERTION_TYPE_FAILED`[#](https://nodejs.org/docs/latest/api/errors.html#err_import_assertion_type_failed)

추가된 버전: v17.1.0, v16.14.0  
제거된 버전: v21.1.0

임포트 어설션(import assertion)이 실패하여 지정된 모듈을 불러오지 못했습니다.


#### `ERR_IMPORT_ASSERTION_TYPE_MISSING`[#](https://nodejs.org/docs/latest/api/errors.html#err_import_assertion_type_missing)

추가된 버전: v17.1.0, v16.14.0  
제거된 버전: v21.1.0

이 오류는 **import assertion**이 누락되어 지정된 모듈을 불러올 수 없을 때 발생합니다.


#### `ERR_IMPORT_ASSERTION_TYPE_UNSUPPORTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_import_assertion_type_unsupported)

추가된 버전: v17.1.0, v16.14.0  
제거된 버전: v21.1.0

이 버전의 Node.js에서는 import 속성을 지원하지 않습니다.


#### `ERR_INDEX_OUT_OF_RANGE`[#](https://nodejs.org/docs/latest/api/errors.html#err_index_out_of_range)

추가된 버전: v10.0.0  
제거된 버전: v11.0.0

지정된 인덱스가 허용된 범위를 벗어났을 때 발생합니다. 예를 들어, 음수 오프셋이 사용된 경우가 이에 해당합니다.


#### `ERR_INVALID_OPT_VALUE`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_opt_value)

추가된 버전: v8.0.0  
제거된 버전: v15.0.0

옵션 객체에 잘못된 값이나 예상치 못한 값이 전달되었을 때 발생하는 오류입니다.


#### `ERR_INVALID_OPT_VALUE_ENCODING`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_opt_value_encoding)

추가된 버전: v9.0.0  
제거된 버전: v15.0.0

유효하지 않거나 알려지지 않은 파일 인코딩이 전달되었을 때 발생하는 오류입니다.


#### `ERR_INVALID_PERFORMANCE_MARK`[#](https://nodejs.org/docs/latest/api/errors.html#err_invalid_performance_mark)

추가된 버전: v8.5.0  
제거된 버전: v16.7.0

Performance Timing API(`perf_hooks`)를 사용할 때, 유효하지 않은 performance mark가 발생한 경우 이 에러가 나타납니다.


#### `ERR_INVALID_TRANSFER_OBJECT`

`postMessage()`에 잘못된 전송 객체가 전달되었습니다.


#### `ERR_MANIFEST_ASSERT_INTEGRITY`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_assert_integrity)

**제거된 버전:** v22.2.0

리소스를 로드하려고 시도했지만, 해당 리소스가 정책 매니페스트에 정의된 무결성과 일치하지 않았습니다. 더 자세한 정보는 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_DEPENDENCY_MISSING`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_dependency_missing)

**제거된 버전:** v22.2.0

리소스를 로드하려고 시도했지만, 해당 리소스가 로드를 시도한 위치에서 의존성으로 등록되어 있지 않았을 때 발생하는 오류입니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_INTEGRITY_MISMATCH`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_integrity_mismatch)

**제거된 버전:** v22.2.0

정책 매니페스트를 로드하려고 시도했지만, 매니페스트에 동일한 리소스에 대한 여러 항목이 서로 일치하지 않았습니다. 이 오류를 해결하려면 매니페스트 항목을 일치하도록 업데이트해야 합니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_INVALID_RESOURCE_FIELD`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_invalid_resource_field)

**제거된 버전:** v22.2.0

정책 매니페스트 리소스의 필드 중 하나에 유효하지 않은 값이 포함되어 있습니다. 이 오류를 해결하려면 매니페스트 항목을 올바르게 업데이트해야 합니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_INVALID_SPECIFIER`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_invalid_specifier)

**제거된 버전:** v22.2.0

정책 매니페스트 리소스의 의존성 매핑 중 하나에 유효하지 않은 값이 포함되어 있습니다. 이 오류를 해결하려면 매니페스트 항목을 업데이트하여 올바른 값과 일치시켜야 합니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_PARSE_POLICY`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_parse_policy)

**제거된 버전:** v22.2.0

정책 매니페스트를 로드하려고 시도했지만, 매니페스트를 파싱할 수 없었습니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MANIFEST_TDZ`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_tdz)

**제거된 버전:** v22.2.0

정책 매니페스트에서 데이터를 읽으려고 시도했지만, 매니페스트 초기화가 아직 이루어지지 않았습니다. 이는 Node.js의 버그일 가능성이 높습니다.


#### `ERR_MANIFEST_UNKNOWN_ONERROR`[#](https://nodejs.org/docs/latest/api/errors.html#err_manifest_unknown_onerror)

**제거된 버전:** v22.2.0

정책 매니페스트가 로드되었지만, "onerror" 동작에 대해 알 수 없는 값이 포함되어 있습니다. 자세한 내용은 정책 매니페스트 문서를 참고하세요.


#### `ERR_MISSING_MESSAGE_PORT_IN_TRANSFER_LIST`[#](https://nodejs.org/docs/latest/api/errors.html#err_missing_message_port_in_transfer_list)

**제거된 버전:** v15.0.0

이 오류 코드는 Node.js v15.0.0에서 [`ERR_MISSING_TRANSFERABLE_IN_TRANSFER_LIST`](https://nodejs.org/docs/latest/api/errors.html#err_missing_transferable_in_transfer_list)로 대체되었습니다. 이는 현재 다른 타입의 전송 가능한 객체들도 존재하기 때문에 더 이상 정확하지 않기 때문입니다.


#### `ERR_MISSING_TRANSFERABLE_IN_TRANSFER_LIST`

`postMessage()` 호출에 전달된 객체 중 `transferList` 인자에 명시적으로 포함되어야 하는 객체가 누락된 경우 발생하는 오류입니다. 일반적으로 이 객체는 `MessagePort`입니다.

Node.js v15.0.0 이전 버전에서는 이 오류 코드가 [`ERR_MISSING_MESSAGE_PORT_IN_TRANSFER_LIST`](https://nodejs.org/docs/latest/api/errors.html#err_missing_message_port_in_transfer_list)로 사용되었습니다. 하지만, `MessagePort` 외에도 더 많은 타입의 전송 가능한 객체를 다루기 위해 오류 코드가 확장되었습니다.


#### `ERR_NAPI_CONS_PROTOTYPE_OBJECT`

**추가된 버전:** v9.0.0  
**제거된 버전:** v10.0.0

이 오류는 `Node-API`에서 `Constructor.prototype`이 객체가 아닐 때 발생합니다.


#### `ERR_NAPI_TSFN_START_IDLE_LOOP`

추가된 버전: v10.6.0, v8.16.0  
제거된 버전: v14.2.0, v12.17.0

메인 스레드에서 스레드 안전 함수(thread-safe function)와 연결된 큐에서 값을 제거할 때, idle loop를 사용합니다. 이 오류는 해당 루프를 시작하려고 시도하는 과정에서 문제가 발생했음을 나타냅니다.


#### `ERR_NAPI_TSFN_STOP_IDLE_LOOP`[#](https://nodejs.org/docs/latest/api/errors.html#err_napi_tsfn_stop_idle_loop)

추가된 버전: v10.6.0, v8.16.0  
제거된 버전: v14.2.0, v12.17.0

큐에 더 이상 아이템이 남아있지 않을 때, idle loop를 중지해야 합니다. 이 오류는 idle loop가 멈추지 못했음을 나타냅니다.


#### `ERR_NO_LONGER_SUPPORTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_no_longer_supported)

Node.js API가 더 이상 지원되지 않는 방식으로 호출되었을 때 발생하는 오류입니다. 예를 들어, `Buffer.write(string, encoding, offset[, length])`와 같은 방식으로 사용하면 이 오류가 발생할 수 있습니다.


#### `ERR_OUTOFMEMORY`[#](https://nodejs.org/docs/latest/api/errors.html#err_outofmemory)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 에러는 메모리 부족 상태를 일으킨 작업을 일반적으로 식별하기 위해 사용됩니다.


#### `ERR_PARSE_HISTORY_DATA`[#](https://nodejs.org/docs/latest/api/errors.html#err_parse_history_data)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

`node:repl` 모듈이 REPL 히스토리 파일의 데이터를 파싱하는 데 실패했습니다.


#### `ERR_SOCKET_CANNOT_SEND`[#](https://nodejs.org/docs/latest/api/errors.html#err_socket_cannot_send)

추가된 버전: v9.0.0  
제거된 버전: v14.0.0

소켓을 통해 데이터를 보낼 수 없을 때 발생하는 오류입니다.


#### `ERR_STDERR_CLOSE`[#](https://nodejs.org/docs/latest/api/errors.html#err_stderr_close)

`process.stderr` 스트림을 닫으려는 시도가 있었습니다. Node.js는 설계상 `stdout`이나 `stderr` 스트림을 사용자 코드로 닫는 것을 허용하지 않습니다.


#### `ERR_STDOUT_CLOSE`[#](https://nodejs.org/docs/latest/api/errors.html#err_stdout_close)

`process.stdout` 스트림을 닫으려는 시도가 있었습니다. Node.js는 설계상 `stdout`이나 `stderr` 스트림을 사용자 코드로 닫는 것을 허용하지 않습니다.


#### `ERR_STREAM_READ_NOT_IMPLEMENTED`[#](https://nodejs.org/docs/latest/api/errors.html#err_stream_read_not_implemented)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 오류는 [`readable._read()`](https://nodejs.org/docs/latest/api/stream.html#readable_readsize) 메서드가 구현되지 않은 읽기 가능한 스트림을 사용하려고 할 때 발생합니다.


#### `ERR_TLS_RENEGOTIATION_FAILED`

**추가된 버전:** v9.0.0  
**제거된 버전:** v10.0.0

이 오류는 TLS 재협상 요청이 특정하지 않은 방식으로 실패했을 때 사용됩니다.


#### `ERR_TRANSFERRING_EXTERNALIZED_SHAREDARRAYBUFFER`[#](https://nodejs.org/docs/latest/api/errors.html#err_transferring_externalized_sharedarraybuffer)

**추가된 버전:** v10.5.0  
**제거된 버전:** v14.0.0

직렬화(serialization) 과정에서 JavaScript 엔진이나 Node.js가 관리하지 않는 메모리를 가진 `SharedArrayBuffer`가 발견되었습니다. 이러한 `SharedArrayBuffer`는 직렬화할 수 없습니다.

이 오류는 네이티브 애드온(native addons)이 "외부화된(externalized)" 모드로 `SharedArrayBuffer`를 생성하거나, 기존의 `SharedArrayBuffer`를 외부화된 모드로 전환할 때만 발생할 수 있습니다.


#### `ERR_UNKNOWN_STDIN_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_unknown_stdin_type)

**추가된 버전:** v8.0.0  
**제거된 버전:** v11.7.0

Node.js 프로세스를 실행할 때 알 수 없는 `stdin` 파일 타입이 사용되면 이 오류가 발생합니다. 이 오류는 일반적으로 Node.js 자체의 버그를 나타내지만, 사용자 코드에 의해 발생할 수도 있습니다.


#### `ERR_UNKNOWN_STREAM_TYPE`[#](https://nodejs.org/docs/latest/api/errors.html#err_unknown_stream_type)

추가된 버전: v8.0.0  
제거된 버전: v11.7.0

Node.js 프로세스를 실행할 때 알 수 없는 `stdout` 또는 `stderr` 파일 타입이 사용되면 이 오류가 발생합니다. 이 오류는 일반적으로 Node.js 자체의 버그를 나타내지만, 사용자 코드에 의해 발생할 수도 있습니다.


#### `ERR_V8BREAKITERATOR`[#](https://nodejs.org/docs/latest/api/errors.html#err_v8breakiterator)

V8의 `BreakIterator` API를 사용했지만, 전체 ICU 데이터 세트가 설치되지 않았을 때 발생하는 오류입니다.


#### `ERR_VALUE_OUT_OF_RANGE`[#](https://nodejs.org/docs/latest/api/errors.html#err_value_out_of_range)

추가된 버전: v9.0.0  
제거된 버전: v10.0.0

이 오류는 주어진 값이 허용된 범위를 벗어났을 때 발생합니다.


#### `ERR_VM_MODULE_LINKING_ERRORED`[#](https://nodejs.org/docs/latest/api/errors.html#err_vm_module_linking_errored)

**추가된 버전:** v10.0.0  
**제거된 버전:** v18.1.0, v16.17.0

링커 함수가 링크에 실패한 모듈을 반환했을 때 발생하는 에러입니다.


#### `ERR_VM_MODULE_NOT_LINKED`

모듈은 인스턴스화되기 전에 반드시 성공적으로 연결되어야 합니다.


#### `ERR_WORKER_UNSUPPORTED_EXTENSION`[#](https://nodejs.org/docs/latest/api/errors.html#err_worker_unsupported_extension)

추가된 버전: v11.0.0  
제거된 버전: v16.9.0

워커의 메인 스크립트로 사용된 경로에 알 수 없는 파일 확장자가 포함되어 있습니다.


#### `ERR_ZLIB_BINDING_CLOSED`

**추가된 버전:** v9.0.0  
**제거된 버전:** v10.0.0

이 오류는 이미 닫힌 `zlib` 객체를 사용하려고 시도할 때 발생합니다.

```javascript
const zlib = require('zlib');
const gzip = zlib.createGzip();

gzip.close();

// 이미 닫힌 gzip 객체를 사용하려고 하면 ERR_ZLIB_BINDING_CLOSED 오류가 발생합니다.
gzip.write('some data');
```

위 예제에서 `gzip` 객체를 닫은 후에 다시 사용하려고 하면 `ERR_ZLIB_BINDING_CLOSED` 오류가 발생합니다. 이 오류는 `zlib` 객체가 더 이상 사용할 수 없음을 나타냅니다.


